<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Desirée Brösamle" />


<title>SS2 (for AG Hefendehl): Integration</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scSeq_Hefendehl</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SS2 (for AG Hefendehl): Integration</h1>
<h4 class="author">Desirée Brösamle</h4>
<h4 class="date">Dec 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>S:/KJP_Biolabor/Projects/scSeq_Hefendehl/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220131code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220131)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220131code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220131)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltree879143430fc93e19a057a248fec891a6970ee201targetblank8791434a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/879143430fc93e19a057a248fec891a6970ee201" target="_blank">8791434</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltree879143430fc93e19a057a248fec891a6970ee201targetblank8791434a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/879143430fc93e19a057a248fec891a6970ee201" target="_blank">8791434</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/ReloadAllData_Hefendehl_Stroke_Dec'21.RData
    Ignored:    data/Sample_Tables/
    Ignored:    data/counts.csv
    Ignored:    data/genecounts.csv
    Ignored:    data/microglia_protein.rds
    Ignored:    data/samples.integrated.RData
    Ignored:    data/tx2genes.csv
    Ignored:    output/Descriptives.Rmd
    Ignored:    output/Descriptives.docx

Untracked files:
    Untracked:  geneviewer/Dataset.RData
    Untracked:  workflow_helper.R

Unstaged changes:
    Modified:   geneviewer/app.R
    Modified:   geneviewer/rsconnect/shinyapps.io/molgenlab/geneviewer.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="preprocessing" class="section level1">
<h1><strong>Preprocessing</strong></h1>
<p><strong>QC metrics from AG Hefendehl (Stroke)</strong></p>
<pre class="r"><code># Visualize QC metrics as a violin plot
Idents(plates_hefendehl) &lt;- &quot;Plate&quot;
VlnPlot(plates_hefendehl, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mito&quot;,&quot;percent.ribo&quot;), 
        group.by = &quot;Plate&quot;, pt.size = 1, ncol = 4, )</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot1 | plot2</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-3-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>QC metrics from AG Neher (own additional WT
control)</strong></p>
<pre class="r"><code># plates_neher &lt;- subset(plates_neher, subset = Genotype == &quot;wt&quot;)
# Visualize QC metrics as a violin plot
Idents(plates_neher) &lt;- &quot;Plate&quot;
VlnPlot(plates_neher, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mito&quot;,&quot;percent.ribo&quot;), 
        group.by = &quot;Plate&quot;,pt.size =1, ncol = 4)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot1a | plot2a</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-4-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Combine both seurat objects and filter cells that have unique feature
counts (gene number) over 7000 or less than 200 and cells with &gt;5%
mitochondrial counts.</p>
<p><strong>QC metrics after filtering</strong></p>
<pre class="r"><code>plot3 | plot4</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="perform-integrated-analysis" class="section level1">
<h1><strong>Perform integrated analysis</strong></h1>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-6-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>PCA, tSNE and UMAP from the first 30 dimensions and with a resolution
of 0.8</p>
<pre class="r"><code># Visualization
Idents(samples.integrated) &lt;- &quot;seurat_clusters&quot;

DimPlot(samples.integrated, reduction = &quot;umap&quot;, split.by = &quot;Treatment&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-7-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1 + p1a +  p1b</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-7-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(samples.integrated, group.by = &quot;seurat_clusters&quot;, pt.size =1, label = T) + NoLegend()</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-7-3.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(samples.integrated, reduction = &quot;umap&quot;, split.by = &quot;Plate&quot;, pt.size = 1)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-7-4.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(samples.integrated, reduction = &quot;umap&quot;, split.by = &quot;Genotype_Treatment&quot;, pt.size =1)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-7-5.png" width="1440" style="display: block; margin: auto;" /></p>
<p><em>Cell cycle scoring</em></p>
<p>First, a score is assigned to each cell (Tirosh et al. 2016), based
on its expression of G2/M and S phase markers. These markers should be
anticorrelated in their expression levels and cells expression neither
are likely not cycling and in G1 phase. Note: For downstream cell cylce
regression the quantitative scores for G2/M and S phase are used, not
the dicrete classification.</p>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20cycle%20scoring-1.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/cell%20cycle%20scoring-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Cluster distribution</strong></p>
<pre class="r"><code>DimPlot(object = samples.integrated, pt.size = 1,reduction = &quot;umap&quot;, group.by=&quot;Age&quot;,label = F) +
  ggtitle(&quot;Cluster distribution according to Age&quot;) | DimPlot(object = samples.integrated, pt.size = 1,reduction = &quot;umap&quot;, group.by=&quot;Genotype_Treatment&quot;,label = F) +
  ggtitle(&quot;Cluster distribution according to Genotype + Treatment&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cluster%20distribution-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(object = samples.integrated, pt.size = 1,reduction = &quot;umap&quot;, group.by=&quot;Plate&quot;,label = F) +
  ggtitle(&quot;Cluster distribution according to Plate&quot;) | DimPlot(object = samples.integrated, pt.size = 1,reduction = &quot;umap&quot;, group.by = &quot;seurat_clusters&quot;, label = T) +
  ggtitle(&quot;Clustering&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cluster%20distribution-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(samples.integrated) &lt;- &quot;seurat_clusters&quot;

# use same colors for clusters as in plots
require(scales)
identities &lt;- levels(samples.integrated$seurat_clusters) # Create vector with levels of object@ident
cluster_colors &lt;- hue_pal()(length(identities)) # Create vector of default ggplot2 colors


# number of cells in each cluster 
cluster_nCell &lt;- as.data.frame.matrix(table(samples.integrated$seurat_clusters,
                                            samples.integrated$Genotype_Age))
cluster_nCell[&quot;Total&quot; ,] = colSums(cluster_nCell)
cluster_nCell</code></pre>
<pre><code>      APPPS1+_10mo APPPS1+_9mo MX04+_10mo MX04+_9mo wt_10mo wt_17mo wt_9mo
0               42          57         20        18      15      20     34
1               23          38          3         7       9      68     16
2               22          29          2         3       2      31     18
3               14          25         12         5       6      29      9
4                4           6          1         0       1      11      3
5                3           4          5         1       5       3      3
6                6          10          0         0       3       0      3
Total          114         169         43        34      41     162     86</code></pre>
<pre class="r"><code># % of cells in each cluster , grouped by genotype_Age
cluster_percent_Cell &lt;- data.frame(round((prop.table(x = table(samples.integrated$seurat_clusters, 
                              samples.integrated$Genotype_Age), margin = 2)*100),2))
colnames(cluster_percent_Cell) &lt;- c(&quot;Genotype_Age&quot;, &quot;Celltype&quot;, &quot;Frequency&quot;)
cluster_percent_Cell</code></pre>
<pre><code>   Genotype_Age     Celltype Frequency
1             0 APPPS1+_10mo     36.84
2             1 APPPS1+_10mo     20.18
3             2 APPPS1+_10mo     19.30
4             3 APPPS1+_10mo     12.28
5             4 APPPS1+_10mo      3.51
6             5 APPPS1+_10mo      2.63
7             6 APPPS1+_10mo      5.26
8             0  APPPS1+_9mo     33.73
9             1  APPPS1+_9mo     22.49
10            2  APPPS1+_9mo     17.16
11            3  APPPS1+_9mo     14.79
12            4  APPPS1+_9mo      3.55
13            5  APPPS1+_9mo      2.37
14            6  APPPS1+_9mo      5.92
15            0   MX04+_10mo     46.51
16            1   MX04+_10mo      6.98
17            2   MX04+_10mo      4.65
18            3   MX04+_10mo     27.91
19            4   MX04+_10mo      2.33
20            5   MX04+_10mo     11.63
21            6   MX04+_10mo      0.00
22            0    MX04+_9mo     52.94
23            1    MX04+_9mo     20.59
24            2    MX04+_9mo      8.82
25            3    MX04+_9mo     14.71
26            4    MX04+_9mo      0.00
27            5    MX04+_9mo      2.94
28            6    MX04+_9mo      0.00
29            0      wt_10mo     36.59
30            1      wt_10mo     21.95
31            2      wt_10mo      4.88
32            3      wt_10mo     14.63
33            4      wt_10mo      2.44
34            5      wt_10mo     12.20
35            6      wt_10mo      7.32
36            0      wt_17mo     12.35
37            1      wt_17mo     41.98
38            2      wt_17mo     19.14
39            3      wt_17mo     17.90
40            4      wt_17mo      6.79
41            5      wt_17mo      1.85
42            6      wt_17mo      0.00
43            0       wt_9mo     39.53
44            1       wt_9mo     18.60
45            2       wt_9mo     20.93
46            3       wt_9mo     10.47
47            4       wt_9mo      3.49
48            5       wt_9mo      3.49
49            6       wt_9mo      3.49</code></pre>
<pre class="r"><code># Grouped barchart of cell proportions
ggplot(cluster_percent_Cell, aes(fill=Genotype_Age, y=Frequency, x=Celltype)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;)+
  ggtitle(&quot;Cell distribution according to Genotype_Age [%]&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=1)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20per%20cluster-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code># % of cells in each cluster , grouped by genotype_treatment
cluster_percent_Cell &lt;- data.frame(round((prop.table(x = table(samples.integrated$seurat_clusters, 
                              samples.integrated$Genotype_Treatment), margin = 2)*100),2))
colnames(cluster_percent_Cell) &lt;- c(&quot;Genotype_Treatment&quot;, &quot;Celltype&quot;, &quot;Frequency&quot;)
cluster_percent_Cell</code></pre>
<pre><code>   Genotype_Treatment       Celltype Frequency
1                   0   APPPS1+_Ctrl     30.32
2                   1   APPPS1+_Ctrl     23.87
3                   2   APPPS1+_Ctrl     21.94
4                   3   APPPS1+_Ctrl     14.84
5                   4   APPPS1+_Ctrl      5.81
6                   5   APPPS1+_Ctrl      1.94
7                   6   APPPS1+_Ctrl      1.29
8                   0 APPPS1+_Stroke     40.62
9                   1 APPPS1+_Stroke     18.75
10                  2 APPPS1+_Stroke     13.28
11                  3 APPPS1+_Stroke     12.50
12                  4 APPPS1+_Stroke      0.78
13                  5 APPPS1+_Stroke      3.12
14                  6 APPPS1+_Stroke     10.94
15                  0     MX04+_Ctrl     50.68
16                  1     MX04+_Ctrl     10.96
17                  2     MX04+_Ctrl      6.85
18                  3     MX04+_Ctrl     21.92
19                  4     MX04+_Ctrl      1.37
20                  5     MX04+_Ctrl      8.22
21                  6     MX04+_Ctrl      0.00
22                  0   MX04+_Stroke     25.00
23                  1   MX04+_Stroke     50.00
24                  2   MX04+_Stroke      0.00
25                  3   MX04+_Stroke     25.00
26                  4   MX04+_Stroke      0.00
27                  5   MX04+_Stroke      0.00
28                  6   MX04+_Stroke      0.00
29                  0        wt_Ctrl     12.35
30                  1        wt_Ctrl     41.98
31                  2        wt_Ctrl     19.14
32                  3        wt_Ctrl     17.90
33                  4        wt_Ctrl      6.79
34                  5        wt_Ctrl      1.85
35                  6        wt_Ctrl      0.00
36                  0      wt_Stroke     38.58
37                  1      wt_Stroke     19.69
38                  2      wt_Stroke     15.75
39                  3      wt_Stroke     11.81
40                  4      wt_Stroke      3.15
41                  5      wt_Stroke      6.30
42                  6      wt_Stroke      4.72</code></pre>
<pre class="r"><code># Grouped barchart of cell proportions
ggplot(cluster_percent_Cell, aes(fill=Genotype_Treatment, y=Frequency, x=Celltype)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;)+
  ggtitle(&quot;Cell distribution according to Genotype_Treatment [%]&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=1)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20per%20cluster-2.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-type-identification" class="section level1">
<h1><strong>Cell type identification</strong></h1>
<div id="finding-differentially-expressed-features-cluster-biomarkers"
class="section level2">
<h2><strong>Finding differentially expressed features (cluster
biomarkers)</strong></h2>
<pre class="r"><code>Idents(samples.integrated) &lt;- samples.integrated$seurat_clusters

DoHeatmap(samples.integrated, features = top20$gene) + NoLegend() + ggtitle(&quot;Top20 cluster marker genes&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/Marker%20genes-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="singler" class="section level2">
<h2><strong>SingleR</strong></h2>
<p>SingleR is an automatic annotation method for (scRNAseq) data (Aran
et al. 2019). Given a reference dataset of samples (single-cell or bulk)
with known labels, it labels new cells from a test dataset based on
similarity to the reference set. Here we use the built-in references
“Immgen” (830 microarray samples of sorted hematopoetic and immune cell
populations) and “Mouse RNA-Seq” (358 non-specific mouse RNA-seq
samples).</p>
<p><strong>Immgen reference</strong></p>
<p>Section Skipped as pred.immgen object is not available</p>
<p><strong>MouseRNA-Seq reference</strong> Section Skipped as
pred.mouseRNA object is not available</p>
<pre class="r"><code># number of cells in each cluster 
cluster_nCell &lt;- data.frame(table(samples.integrated$MouseRNASeq_sc_labels,samples.integrated$Genotype_Treatment))
colnames(cluster_nCell) &lt;- c(&quot;Genotype_Treatment&quot;, &quot;MouseRNASeq_sc_labels&quot;, &quot;Number&quot;)

#cluster_nCell[&quot;Total&quot; ,] = colSums(cluster_nCell)
# cluster_nCell

# % of cells in each cluster 
cluster_percent_Cell &lt;- data.frame(round((prop.table(x = table(samples.integrated$MouseRNASeq_sc_labels, 
                              samples.integrated$Genotype_Treatment), margin = 2)*100),2))
colnames(cluster_percent_Cell) &lt;- c(&quot;Genotype_Treatment&quot;, &quot;MouseRNASeq_sc_labels&quot;, &quot;Frequency&quot;)
# cluster_percent_Cell


# Grouped barchart of absolute cell numbers
ggplot(cluster_nCell, aes(fill=Genotype_Treatment, y=Number, x=MouseRNASeq_sc_labels)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;) +
  ggtitle(&quot;Cell distribution according to MouseRNASeq reference (absolut values)&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=0.5)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20MouseRNASeq-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Grouped barchart of cell proportions
ggplot(cluster_percent_Cell, aes(fill=Genotype_Treatment, y=Frequency, x=MouseRNASeq_sc_labels)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;)+
  ggtitle(&quot;Cell distribution according to MouseRNASeq reference [%]&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=0.5)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20MouseRNASeq-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code># number of cells in each cluster 
cluster_nCell &lt;- data.frame(table(samples.integrated$Immgen_sc_labels,samples.integrated$Treatment))
colnames(cluster_nCell) &lt;- c(&quot;Treatment&quot;, &quot;Immgen_sc_labels&quot;, &quot;Number&quot;)

#cluster_nCell[&quot;Total&quot; ,] = colSums(cluster_nCell)
# cluster_nCell

# % of cells in each cluster 
cluster_percent_Cell &lt;- data.frame(round((prop.table(x = table(samples.integrated$Immgen_sc_labels, 
                              samples.integrated$Treatment), margin = 2)*100),2))
colnames(cluster_percent_Cell) &lt;- c(&quot;Treatment&quot;, &quot;Immgen_sc_labels&quot;, &quot;Frequency&quot;)
# cluster_percent_Cell


# Grouped barchart of absolute cell numbers
ggplot(cluster_nCell, aes(fill=Treatment, y=Number, x=Immgen_sc_labels)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;) +
  ggtitle(&quot;Cell distribution according to Immgen reference (absolut values)&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=0.5)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20immgen-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Grouped barchart of cell proportions
ggplot(cluster_percent_Cell, aes(fill=Treatment, y=Frequency, x=Immgen_sc_labels)) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;)+
  ggtitle(&quot;Cell distribution according to Immgen reference [%]&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=0.5)) +
  xlab(&quot;&quot;)</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cell%20distribution%20immgen-2.png" width="1440" style="display: block; margin: auto;" />
Renaming Clusters</p>
<pre class="r"><code>DimPlot(samples.integrated, reduction = &quot;umap&quot;, label=T) + NoLegend()</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="zoom-into-microglia-clusters" class="section level1">
<h1><strong>Zoom into Microglia clusters</strong></h1>
<p>A second Seurat cluster is generated, where all non-microglial immune
cells are removed according to the mouseRNASeq_sc_labels.</p>
<pre class="r"><code>p3 + p3a</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/subclustering%20microglia-1.png" width="1440" style="display: block; margin: auto;" /></p>
<div
id="dams-cell.-2017-jun-1516971276-1290.e17.-doi-10.1016j.cell.2017.05.018"
class="section level2">
<h2><strong>DAMs (Cell. 2017 Jun 15;169(7):1276-1290.e17. doi:
10.1016/j.cell.2017.05.018)</strong></h2>
<p><strong>DAM plotting</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/Stage%202%20DAMs-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p5 | p5a</code></pre>
<pre><code>Warning: Use of `expr_df[[axes[1]]]` is discouraged. Use `.data[[axes[1]]]`
instead.</code></pre>
<pre><code>Warning: Use of `expr_df[[axes[2]]]` is discouraged. Use `.data[[axes[2]]]`
instead.</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-1.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-2.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-3.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-4.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-5.png" width="1440" style="display: block; margin: auto;" /><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-10-6.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="renaming-clusters" class="section level2">
<h2><strong>Renaming clusters</strong></h2>
<pre class="r"><code>Idents(microglia) &lt;- &quot;Celltype&quot;
DimPlot(microglia, reduction = &quot;umap&quot;, label = TRUE, pt.size = 1)+ NoLegend()</code></pre>
<p><img src="figure/00_Original_Analysis.Rmd/cluster%20renaming-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="signature-enrichment" class="section level2">
<h2><strong>Signature enrichment</strong></h2>
<div id="degs-and-gsea-between-microglia-of-different-genotypes"
class="section level3">
<h3><strong>DEGs and GSEA between microglia of different
genotypes</strong></h3>
<p><strong>WT vs App</strong></p>
<p><strong>WT vs MX04+</strong></p>
<p><strong>MX04+ vs App</strong></p>
</div>
<div id="degs-and-gsea-of-microglia-of-different-genotype-and-treatment"
class="section level3">
<h3><strong>DEGs and GSEA of microglia of different genotype and
treatment</strong></h3>
<p><strong>WT_Ctrl vs WT_Stroke</strong></p>
<p><strong>APP_Ctrl vs APP_Stroke</strong></p>
<p><strong>WT_Ctrl vs APP_Ctrl</strong></p>
<p><strong>WT_Stroke vs APP_Stroke</strong></p>
</div>
<div id="degs-and-gsea-of-each-cluster" class="section level3">
<h3><strong>DEGs and GSEA of each cluster</strong></h3>
<p><strong>Microglia_0</strong></p>
<p>Only works if ReloadAllData_Harmony_Hefendehl_Stroke.RData is loaded.
Otherwise, an error message will occur.</p>
<p><strong>Microglia_1</strong></p>
<p><strong>Microglia_2</strong></p>
<p><strong>Microglia_4</strong></p>
<p><strong>DAM</strong></p>
<p><strong>Microglia_5</strong></p>
<p>Only works if ReloadAllData_Harmony_Hefendehl_Stroke.RData is loaded.
Otherwise, an error message will occur.</p>
</div>
</div>
<div id="genotype-treatment-comparison-in-each-cluster"
class="section level2">
<h2>Genotype + Treatment comparison in each cluster</h2>
<p><strong>Microglia_0</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Microglia_1</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Microglia_2</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>DAM?</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-24-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Microglia_4</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-25-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Microglia_5</strong></p>
<p><img src="figure/00_Original_Analysis.Rmd/unnamed-chunk-26-1.png" width="1440" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scales_1.2.0                pheatmap_1.0.12            
 [3] EnhancedVolcano_1.12.0      ggrepel_0.9.1              
 [5] SingleR_1.10.0              SummarizedExperiment_1.26.1
 [7] Biobase_2.56.0              GenomicRanges_1.48.0       
 [9] GenomeInfoDb_1.32.1         IRanges_2.30.0             
[11] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[13] MatrixGenerics_1.8.0        matrixStats_0.62.0         
[15] ggplot2_3.3.6               sp_1.4-7                   
[17] SeuratObject_4.1.0          Seurat_4.1.1               
[19] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                reticulate_1.25          
  [3] tidyselect_1.1.2          htmlwidgets_1.5.4        
  [5] grid_4.1.3                BiocParallel_1.30.0      
  [7] Rtsne_0.16                munsell_0.5.0            
  [9] ScaledMatrix_1.4.0        codetools_0.2-18         
 [11] ica_1.0-2                 future_1.25.0            
 [13] miniUI_0.1.1.1            withr_2.5.0              
 [15] spatstat.random_2.2-0     colorspace_2.0-3         
 [17] progressr_0.10.0          highr_0.9                
 [19] ggalt_0.4.0               knitr_1.39               
 [21] rstudioapi_0.13           ROCR_1.0-11              
 [23] tensor_1.5                Rttf2pt1_1.3.10          
 [25] listenv_0.8.0             labeling_0.4.2           
 [27] git2r_0.30.1              GenomeInfoDbData_1.2.8   
 [29] polyclip_1.10-0           farver_2.1.0             
 [31] rprojroot_2.0.3           parallelly_1.31.1        
 [33] vctrs_0.4.1               generics_0.1.2           
 [35] xfun_0.31                 R6_2.5.1                 
 [37] ggbeeswarm_0.6.0          rsvd_1.0.5               
 [39] bitops_1.0-7              spatstat.utils_2.3-1     
 [41] DelayedArray_0.22.0       assertthat_0.2.1         
 [43] promises_1.2.0.1          rgeos_0.5-9              
 [45] beeswarm_0.4.0            gtable_0.3.0             
 [47] beachmat_2.12.0           ash_1.0-15               
 [49] globals_0.15.0            processx_3.5.3           
 [51] goftest_1.2-3             rlang_1.0.2              
 [53] splines_4.1.3             extrafontdb_1.0          
 [55] lazyeval_0.2.2            spatstat.geom_2.4-0      
 [57] yaml_2.3.5                reshape2_1.4.4           
 [59] abind_1.4-5               httpuv_1.6.5             
 [61] extrafont_0.18            tools_4.1.3              
 [63] ellipsis_0.3.2            spatstat.core_2.4-2      
 [65] jquerylib_0.1.4           RColorBrewer_1.1-3       
 [67] ggridges_0.5.3            Rcpp_1.0.8.3             
 [69] plyr_1.8.7                sparseMatrixStats_1.8.0  
 [71] zlibbioc_1.42.0           purrr_0.3.4              
 [73] RCurl_1.98-1.6            ps_1.7.0                 
 [75] rpart_4.1.16              deldir_1.0-6             
 [77] pbapply_1.5-0             cowplot_1.1.1            
 [79] zoo_1.8-10                cluster_2.1.2            
 [81] fs_1.5.2                  magrittr_2.0.3           
 [83] data.table_1.14.2         scattermore_0.8          
 [85] lmtest_0.9-40             RANN_2.6.1               
 [87] whisker_0.4               fitdistrplus_1.1-8       
 [89] patchwork_1.1.1           mime_0.12                
 [91] evaluate_0.15             xtable_1.8-4             
 [93] gridExtra_2.3             compiler_4.1.3           
 [95] tibble_3.1.7              maps_3.4.0               
 [97] KernSmooth_2.23-20        crayon_1.5.1             
 [99] htmltools_0.5.2           mgcv_1.8-39              
[101] later_1.3.0               tidyr_1.2.0              
[103] DBI_1.1.2                 proj4_1.0-11             
[105] MASS_7.3-55               Matrix_1.4-0             
[107] cli_3.3.0                 parallel_4.1.3           
[109] igraph_1.3.1              pkgconfig_2.0.3          
[111] getPass_0.2-2             plotly_4.10.0            
[113] spatstat.sparse_2.1-1     vipor_0.4.5              
[115] bslib_0.3.1               XVector_0.36.0           
[117] stringr_1.4.0             callr_3.7.0              
[119] digest_0.6.29             sctransform_0.3.3        
[121] RcppAnnoy_0.0.19          spatstat.data_2.2-0      
[123] rmarkdown_2.14            leiden_0.4.2             
[125] uwot_0.1.11               DelayedMatrixStats_1.18.0
[127] shiny_1.7.1               lifecycle_1.0.1          
[129] nlme_3.1-155              jsonlite_1.8.0           
[131] BiocNeighbors_1.14.0      viridisLite_0.4.0        
[133] fansi_1.0.3               pillar_1.7.0             
[135] lattice_0.20-45           ggrastr_1.0.1            
[137] fastmap_1.1.0             httr_1.4.3               
[139] survival_3.2-13           glue_1.6.2               
[141] png_0.1-7                 stringi_1.7.6            
[143] sass_0.4.1                BiocSingular_1.12.0      
[145] dplyr_1.0.9               irlba_2.3.5              
[147] future.apply_1.9.0       </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
