<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AGC" />


<title>Biostatistical analysis part 2</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.5.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scSeq_Hefendehl</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Biostatistical analysis part 2</h1>
<h4 class="author">AGC</h4>
<h4 class="date">2 2 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>S:/KJP_Biolabor/Projects/scSeq_Hefendehl/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220131code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220131)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220131code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220131)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltreeef56d059cf3f209b109302a7099676d9ecc3308atargetblankef56d05a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/ef56d059cf3f209b109302a7099676d9ecc3308a" target="_blank">ef56d05</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltreeef56d059cf3f209b109302a7099676d9ecc3308atargetblankef56d05a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/ef56d059cf3f209b109302a7099676d9ecc3308a" target="_blank">ef56d05</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/ReloadAllData_Hefendehl_Stroke_Dec'21.RData
    Ignored:    data/Sample_Tables/
    Ignored:    data/counts.csv
    Ignored:    data/genecounts.csv
    Ignored:    data/microglia_protein.rds
    Ignored:    data/samples.integrated.RData
    Ignored:    data/tx2genes.csv
    Ignored:    output/Descriptives.Rmd
    Ignored:    output/Descriptives.docx

Untracked files:
    Untracked:  app/
    Untracked:  geneviewer/
    Untracked:  genplotter/
    Untracked:  workflow_helper.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#get sample data
samples.integrated@meta.data %&gt;% as.data.frame() -&gt; samplemeta

# convert to correct data type 

# define genotype as is
samplemeta$Genotype_corr = factor(samplemeta$Genotype==&quot;wt&quot;, levels=c(F,T), labels = c(&quot;APPPS1+&quot;, &quot;WT&quot;))
samplemeta$Genotype_corr = relevel(samplemeta$Genotype_corr, ref=&quot;WT&quot;)

samplemeta$methoxy = factor(samplemeta$Genotype==&quot;MX04+&quot;, levels=c(T,F), labels = c(&quot;MX04+&quot;, &quot;MX04-&quot;))

samplemeta$Treatment = as.factor(samplemeta$Treatment)
samplemeta$Treatment = relevel(samplemeta$Treatment, ref=&quot;Ctrl&quot;)
samplemeta$Mouse_ID = as.factor(samplemeta$Mouse_ID)
samplemeta$Sex = as.factor(samplemeta$Sex)
samplemeta$Brain_region = as.factor(samplemeta$Brain_region)
samplemeta$Celltype = as.factor(samplemeta$Celltype)

nCells=nrow(samplemeta)
nMice=nlevels(samplemeta$Mouse_ID)
nCelltypes=nlevels(samplemeta$Celltype)</code></pre>
<div id="sample-descriptive" class="section level2">
<h2>Sample descriptive:</h2>
<p>Data contains a total of 649 Cells from 9. <em>Q:</em> Original raw datset containing only frankfurt data included 1149 cells. What were the filter criteria in the primary cell type analysis</p>
<p><strong>Cells per Mouse</strong></p>
<p>Cells per Strain</p>
<pre class="r"><code>table(Mouse_ID=samplemeta$Mouse_ID) %&gt;% as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Mouse_ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
23#15773
</td>
<td style="text-align:left;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
23#15774
</td>
<td style="text-align:left;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
23#15792
</td>
<td style="text-align:left;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
386
</td>
<td style="text-align:left;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
387
</td>
<td style="text-align:left;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
388
</td>
<td style="text-align:left;">
113
</td>
</tr>
<tr>
<td style="text-align:left;">
409
</td>
<td style="text-align:left;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
457
</td>
<td style="text-align:left;">
115
</td>
</tr>
<tr>
<td style="text-align:left;">
461
</td>
<td style="text-align:left;">
88
</td>
</tr>
</tbody>
</table>
</div>
<p>Cells per Genotype</p>
<pre class="r"><code>table(Genotype=samplemeta$Genotype_corr, Treatment=samplemeta$Treatment) %&gt;% as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Genotype
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Treatment
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Ctrl
</td>
<td style="text-align:left;">
162
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Ctrl
</td>
<td style="text-align:left;">
228
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Stroke
</td>
<td style="text-align:left;">
127
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Stroke
</td>
<td style="text-align:left;">
132
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>table(Genotype=samplemeta$Genotype_corr, Celltype=samplemeta$Celltype) %&gt;% 
  as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Genotype
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Celltype
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
T/NK
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
T/NK
</td>
<td style="text-align:left;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_0
</td>
<td style="text-align:left;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_0
</td>
<td style="text-align:left;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_1
</td>
<td style="text-align:left;">
93
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_1
</td>
<td style="text-align:left;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_2
</td>
<td style="text-align:left;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_2
</td>
<td style="text-align:left;">
56
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_3
</td>
<td style="text-align:left;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_3
</td>
<td style="text-align:left;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_4
</td>
<td style="text-align:left;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_4
</td>
<td style="text-align:left;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_5
</td>
<td style="text-align:left;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_5
</td>
<td style="text-align:left;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Granulocytes
</td>
<td style="text-align:left;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Granulocytes
</td>
<td style="text-align:left;">
16
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>variables=c(&quot;Celltype&quot;,&quot;Sex&quot;, &quot;Genotype_corr&quot;, &quot;Treatment&quot;,&quot;Phase&quot;, &quot;Brain_region&quot;,&quot;nCount_RNA&quot;,&quot;pseudoaligned_reads&quot;, &quot;percent.mito&quot;, &quot;percent.ribo&quot;, &quot;Mouse_ID&quot;)</code></pre>
<p>Descriptive stats across Cell type</p>
<pre class="r"><code>res = compareGroups(Celltype~., data = samplemeta[,variables], max.ylev = 10)
#summary(res)
export_table &lt;- createTable(res)
options(width = 10000)
print(export_table)</code></pre>
<pre><code>
--------Summary descriptives table by &#39;Celltype&#39;---------

_____________________________________________________________________________________________________________________________________________________ 
                         T/NK       Microglia_0    Microglia_1    Microglia_2    Microglia_3    Microglia_4    Microglia_5    Granulocytes  p.overall 
                         N=22          N=186          N=164          N=107           N=99           N=26           N=23           N=22                
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
Sex:                                                                                                                                            .     
    f                 6 (27.3%)      27 (14.5%)     21 (12.8%)     11 (10.3%)     11 (11.1%)     1 (3.85%)      2 (8.70%)      8 (36.4%)              
    m                 16 (72.7%)    159 (85.5%)    143 (87.2%)     96 (89.7%)     88 (88.9%)     25 (96.2%)     21 (91.3%)     14 (63.6%)             
Genotype_corr:                                                                                                                               &lt;0.001   
    WT                3 (13.6%)      66 (35.5%)     93 (56.7%)     51 (47.7%)     44 (44.4%)     15 (57.7%)     11 (47.8%)     6 (27.3%)              
    APPPS1+           19 (86.4%)    120 (64.5%)     71 (43.3%)     56 (52.3%)     55 (55.6%)     11 (42.3%)     12 (52.2%)     16 (72.7%)             
Treatment:                                                                                                                                   &lt;0.001   
    Ctrl              3 (13.6%)     101 (54.3%)    113 (68.9%)     70 (65.4%)     68 (68.7%)     21 (80.8%)     12 (52.2%)     2 (9.09%)              
    Stroke            19 (86.4%)     85 (45.7%)     51 (31.1%)     37 (34.6%)     31 (31.3%)     5 (19.2%)      11 (47.8%)     20 (90.9%)             
Phase:                                                                                                                                        0.002   
    G1                1 (4.55%)      79 (42.5%)     70 (42.7%)     59 (55.1%)     42 (42.4%)     12 (46.2%)     14 (60.9%)     5 (22.7%)              
    G2M               11 (50.0%)     50 (26.9%)     38 (23.2%)     17 (15.9%)     21 (21.2%)     6 (23.1%)      4 (17.4%)      11 (50.0%)             
    S                 10 (45.5%)     57 (30.6%)     56 (34.1%)     31 (29.0%)     36 (36.4%)     8 (30.8%)      5 (21.7%)      6 (27.3%)              
Brain_region:                                                                                                                                &lt;0.001   
    Cortex            3 (13.6%)     101 (54.3%)    113 (68.9%)     70 (65.4%)     68 (68.7%)     21 (80.8%)     12 (52.2%)     2 (9.09%)              
    Lesion            19 (86.4%)     85 (45.7%)     51 (31.1%)     37 (34.6%)     31 (31.3%)     5 (19.2%)      11 (47.8%)     20 (90.9%)             
nCount_RNA          164049 (80916) 150451 (78955) 134224 (54802) 170420 (76702) 152640 (72678) 172200 (70772) 176087 (64299) 144828 (65899)   0.002   
pseudoaligned_reads 165078 (81748) 150942 (78850) 134355 (54810) 170750 (76639) 152790 (72668) 172433 (70717) 176390 (64225) 145316 (65972)   0.002   
percent.mito         2.23 (0.90)    1.75 (1.17)    1.62 (1.01)    2.07 (1.07)    1.72 (1.15)    1.97 (1.02)    2.11 (0.77)    0.92 (1.00)    &lt;0.001   
percent.ribo         6.53 (2.72)    2.68 (1.58)    2.75 (1.81)    2.44 (1.24)    3.28 (1.67)    2.11 (0.85)    2.91 (1.35)    2.01 (1.17)    &lt;0.001   
Mouse_ID:                                                                                                                                       .     
    23#15773          0 (0.00%)      7 (3.76%)      25 (15.2%)     12 (11.2%)     17 (17.2%)     4 (15.4%)      1 (4.35%)      0 (0.00%)              
    23#15774          0 (0.00%)      1 (0.54%)      30 (18.3%)     10 (9.35%)     6 (6.06%)      1 (3.85%)      1 (4.35%)      0 (0.00%)              
    23#15792          1 (4.55%)      11 (5.91%)     13 (7.93%)     9 (8.41%)      6 (6.06%)      6 (23.1%)      1 (4.35%)      0 (0.00%)              
    386               1 (4.55%)      14 (7.53%)     9 (5.49%)      2 (1.87%)      6 (6.06%)      1 (3.85%)      5 (21.7%)      3 (13.6%)              
    387               11 (50.0%)     10 (5.38%)     5 (3.05%)      6 (5.61%)      5 (5.05%)      0 (0.00%)      1 (4.35%)      6 (27.3%)              
    388               1 (4.55%)      41 (22.0%)     21 (12.8%)     18 (16.8%)     21 (21.2%)     5 (19.2%)      6 (26.1%)      0 (0.00%)              
    409               1 (4.55%)      33 (17.7%)     16 (9.76%)     18 (16.8%)     9 (9.09%)      3 (11.5%)      3 (13.0%)      3 (13.6%)              
    457               1 (4.55%)      41 (22.0%)     24 (14.6%)     21 (19.6%)     18 (18.2%)     5 (19.2%)      3 (13.0%)      2 (9.09%)              
    461               6 (27.3%)      28 (15.1%)     21 (12.8%)     11 (10.3%)     11 (11.1%)     1 (3.85%)      2 (8.70%)      8 (36.4%)              
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<pre class="r"><code>export2xls(export_table,paste0(home,&quot;/docs/Descriptives.xlsx&quot;))</code></pre>
<p>download data as excel file <a href="Descriptives.xlsx">here</a></p>
<pre class="r"><code>#get normalized counts 
# question to Desiree hat the Seurat object been initialized with normalized data?
counts &lt;- samples.integrated@assays$RNA@counts %&gt;% as.data.frame()

# drop no variance data and sort by samplemeta
counts &lt;- counts[apply(counts,1, sd) &gt; 0, rownames(samplemeta)]

# drop genes with low detection rate (more than 5 counts per cell)
counts_per_celltype=apply(counts, 1, function(x){tapply(x, samplemeta$Celltype, function(z){sum(z&gt;5,na.rm=T)})})

# keep RNAs with at least 10 cells with goood expression
idx=which(colSums(counts_per_celltype)&gt;10)

counts = counts[idx,]</code></pre>
</div>
<div id="hierarchical-clustering" class="section level2">
<h2>Hierarchical clustering</h2>
<p>to check where the variance in the data comes from</p>
<pre class="r"><code>log2_cpm = log2(counts+1)

varsset=apply(log2_cpm, 1, var)

cpm.sel.trans = t(log2_cpm[order(varsset,decreasing = T)[1:2000],])

distance = dist(cpm.sel.trans)

sampleDistMatrix &lt;- as.matrix(distance)

#colors for plotting heatmap
colors &lt;- rev(colorRampPalette(brewer.pal(9, &quot;Spectral&quot;))(255))
colors=jetcolors(255)
colors=viridis(255)

cellcol = Dark8[1:nlevels(samplemeta$Celltype)]
names(cellcol) = levels(samplemeta$Celltype)

genotypecol = brewer.pal(4,&quot;Accent&quot;)[c(1:nlevels(samplemeta$Genotype_corr))]
names(genotypecol) = levels(samplemeta$Genotype_corr)

strokecol = brewer.pal(5,&quot;Set2&quot;)[1:nlevels(samplemeta$Treatment)+2]
names(strokecol) = levels(samplemeta$Treatment)

mousecol = brewer.pal(9,&quot;Set1&quot;)[1:nlevels(samplemeta$Mouse_ID)]
names(mousecol) = levels(samplemeta$Mouse_ID)

braincol = brewer.pal(3,&quot;Set2&quot;)[1:nlevels(samplemeta$Brain_region)]
names(braincol) = levels(samplemeta$Brain_region)

ann_colors = list(
  Genotype_corr = genotypecol, 
  Mouse_ID = mousecol,
  Brain_region = braincol,
  Celltype=cellcol,
  Treatment=strokecol
)

labels = samplemeta[,c(&quot;Genotype_corr&quot;,&quot;Mouse_ID&quot;, &quot;Brain_region&quot;, &quot;Celltype&quot;, &quot;Treatment&quot;)] %&gt;%  
  mutate_all(as.character) %&gt;% as.data.frame()

rownames(labels)=rownames(samplemeta)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = distance,
         clustering_distance_cols = distance,
         clustering_method = &quot;ward.D2&quot;,
         scale =&quot;none&quot;,
         show_rownames=F, show_colnames = F,
         legend=T,
         border_color = NA, 
         annotation_row = labels,
         annotation_col = labels,
         annotation_colors = ann_colors,
         col = colors, 
         main = &quot;D62 Distances normalized log2 counts&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/basic%20QC-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="genotype-x-treatment-statistical-modelling" class="section level2">
<h2>Genotype X Treatment Statistical modelling</h2>
<pre class="r"><code>getres=function(Celltype=&quot;Specify&quot;, 
                Hypothesis=&quot;~1+Genotype_corr*Treatment&quot;, 
                Target=&quot;Genotype_corrAPPPS1+:TreatmentStroke&quot;,
                Randomeffect=&quot;Sex&quot;){
  
  res= comparison_rand(designform =Hypothesis, 
                       randomeffect = Randomeffect, 
                       Samples = samplemeta$Celltype==Celltype, 
                       log_cpm = log2_cpm, 
                       samplesdata = samplemeta,
                       target=Target)
  
  
  
  labels = samplemeta[samplemeta$Celltype==Celltype,c(&quot;Treatment&quot;, &quot;Genotype_corr&quot;,&quot;Mouse_ID&quot;, &quot;Brain_region&quot;, &quot;Celltype&quot;)] %&gt;%  
    mutate_all(as.character) %&gt;% as.data.frame()
  
  labels=labels %&gt;% arrange(Treatment,Genotype_corr)
  plotdata=log2_cpm[res$adj.P.Val&lt;0.05,rownames(labels)]
  
  pheatmap(plotdata,
           #clustering_method = &quot;ward.D2&quot;,
           cluster_cols = F,
           cluster_rows  = T,
           scale =&quot;column&quot;,
           show_rownames=F, show_colnames = F,
           legend=T,
           border_color = NA, 
           #annotation_row = labels,
           annotation_col = labels,
           annotation_colors = ann_colors,
           col = colors, 
           main = &quot;D62 Distances normalized log2 counts&quot;)
  return(res)
  
}

generate_output=function(CT=&quot;Celltype&quot;, ...){
  respath=paste0(home, &quot;/docs/LMER_&quot;,CT,&quot;.xlsx&quot;)
  
  analysis = getres(Celltype = CT)
  analysis_sig = analysis[analysis$adj.P.Val&lt;0.05,]
  analysis_sig %&gt;% display_tab()
  
  write.xlsx2(analysis_sig, file =respath , sheetName = &quot;significant genes&quot;)
  
  ResGO = getGOresults(rownames(analysis_sig),
                       rownames(analysis),
                       &quot;mmusculus&quot;)
  
  if(length(ResGO)&gt;0){
    p=gostplot(ResGO)
  } else{
    ResGO=data.frame(result=&quot;no significant enrichment identified&quot;)
    p=&quot;no significant enrichment identified&quot;
  }
  
  write.xlsx2(ResGO$result, file = respath, sheetName = &quot;GO_enrichment&quot;, append=T)
  return(list(results=analysis,results_sig=analysis_sig, plot=p))
}</code></pre>
<div id="microglia_0" class="section level3">
<h3>Microglia_0</h3>
<pre class="r"><code>res_output=generate_output(&quot;Microglia_0&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/Microglia_0-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
logFC
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
AveExpr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
t
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
P.Value
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
adj.P.Val
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Tbata
</td>
<td style="text-align:left;">
3.522
</td>
<td style="text-align:left;">
0.454
</td>
<td style="text-align:left;">
14.822
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
60.871
</td>
</tr>
<tr>
<td style="text-align:left;">
AC099934.1
</td>
<td style="text-align:left;">
2.090
</td>
<td style="text-align:left;">
0.223
</td>
<td style="text-align:left;">
13.572
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
53.259
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s1
</td>
<td style="text-align:left;">
4.868
</td>
<td style="text-align:left;">
0.932
</td>
<td style="text-align:left;">
13.345
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
51.865
</td>
</tr>
<tr>
<td style="text-align:left;">
AY036118
</td>
<td style="text-align:left;">
-9.974
</td>
<td style="text-align:left;">
6.541
</td>
<td style="text-align:left;">
-10.095
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
31.986
</td>
</tr>
<tr>
<td style="text-align:left;">
Prrx1
</td>
<td style="text-align:left;">
2.331
</td>
<td style="text-align:left;">
0.394
</td>
<td style="text-align:left;">
9.888
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
30.743
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s2
</td>
<td style="text-align:left;">
3.782
</td>
<td style="text-align:left;">
0.821
</td>
<td style="text-align:left;">
9.270
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
27.086
</td>
</tr>
<tr>
<td style="text-align:left;">
Tmem170
</td>
<td style="text-align:left;">
2.199
</td>
<td style="text-align:left;">
0.349
</td>
<td style="text-align:left;">
9.155
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
26.415
</td>
</tr>
<tr>
<td style="text-align:left;">
Mgmt
</td>
<td style="text-align:left;">
2.584
</td>
<td style="text-align:left;">
0.444
</td>
<td style="text-align:left;">
7.476
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
16.971
</td>
</tr>
<tr>
<td style="text-align:left;">
Lgals3bp
</td>
<td style="text-align:left;">
-7.696
</td>
<td style="text-align:left;">
3.952
</td>
<td style="text-align:left;">
-6.731
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
13.081
</td>
</tr>
<tr>
<td style="text-align:left;">
Olfr907
</td>
<td style="text-align:left;">
2.069
</td>
<td style="text-align:left;">
0.908
</td>
<td style="text-align:left;">
6.403
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
11.441
</td>
</tr>
<tr>
<td style="text-align:left;">
Dock10
</td>
<td style="text-align:left;">
3.245
</td>
<td style="text-align:left;">
5.039
</td>
<td style="text-align:left;">
6.041
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
9.691
</td>
</tr>
<tr>
<td style="text-align:left;">
Myl2
</td>
<td style="text-align:left;">
1.632
</td>
<td style="text-align:left;">
0.203
</td>
<td style="text-align:left;">
6.030
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
9.642
</td>
</tr>
<tr>
<td style="text-align:left;">
Cd52
</td>
<td style="text-align:left;">
-5.507
</td>
<td style="text-align:left;">
3.497
</td>
<td style="text-align:left;">
-5.942
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
9.226
</td>
</tr>
<tr>
<td style="text-align:left;">
Xist
</td>
<td style="text-align:left;">
3.006
</td>
<td style="text-align:left;">
0.748
</td>
<td style="text-align:left;">
5.744
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
8.306
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm26870
</td>
<td style="text-align:left;">
-3.903
</td>
<td style="text-align:left;">
9.134
</td>
<td style="text-align:left;">
-5.639
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
7.830
</td>
</tr>
<tr>
<td style="text-align:left;">
Fhl3
</td>
<td style="text-align:left;">
2.000
</td>
<td style="text-align:left;">
0.607
</td>
<td style="text-align:left;">
5.469
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
7.066
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10076
</td>
<td style="text-align:left;">
-3.831
</td>
<td style="text-align:left;">
2.856
</td>
<td style="text-align:left;">
-5.441
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
6.942
</td>
</tr>
<tr>
<td style="text-align:left;">
Pola2
</td>
<td style="text-align:left;">
-2.445
</td>
<td style="text-align:left;">
2.264
</td>
<td style="text-align:left;">
-5.344
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
6.519
</td>
</tr>
<tr>
<td style="text-align:left;">
Lyz2
</td>
<td style="text-align:left;">
-6.938
</td>
<td style="text-align:left;">
5.945
</td>
<td style="text-align:left;">
-5.214
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
5.958
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10719
</td>
<td style="text-align:left;">
-3.401
</td>
<td style="text-align:left;">
6.711
</td>
<td style="text-align:left;">
-5.214
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
5.957
</td>
</tr>
<tr>
<td style="text-align:left;">
Upk1b
</td>
<td style="text-align:left;">
3.244
</td>
<td style="text-align:left;">
1.015
</td>
<td style="text-align:left;">
5.173
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
5.782
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm15564
</td>
<td style="text-align:left;">
1.530
</td>
<td style="text-align:left;">
0.765
</td>
<td style="text-align:left;">
5.104
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
5.489
</td>
</tr>
<tr>
<td style="text-align:left;">
Col6a3
</td>
<td style="text-align:left;">
2.052
</td>
<td style="text-align:left;">
0.420
</td>
<td style="text-align:left;">
5.023
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
5.150
</td>
</tr>
<tr>
<td style="text-align:left;">
Apoe
</td>
<td style="text-align:left;">
-8.085
</td>
<td style="text-align:left;">
7.561
</td>
<td style="text-align:left;">
-4.941
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
4.811
</td>
</tr>
<tr>
<td style="text-align:left;">
Gnaz
</td>
<td style="text-align:left;">
1.777
</td>
<td style="text-align:left;">
0.780
</td>
<td style="text-align:left;">
4.921
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
4.728
</td>
</tr>
<tr>
<td style="text-align:left;">
D10Wsu102e
</td>
<td style="text-align:left;">
-1.853
</td>
<td style="text-align:left;">
7.009
</td>
<td style="text-align:left;">
-4.852
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
4.445
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm26905
</td>
<td style="text-align:left;">
-3.003
</td>
<td style="text-align:left;">
6.505
</td>
<td style="text-align:left;">
-4.825
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
4.339
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm44215
</td>
<td style="text-align:left;">
-3.369
</td>
<td style="text-align:left;">
3.605
</td>
<td style="text-align:left;">
-4.801
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
4.240
</td>
</tr>
<tr>
<td style="text-align:left;">
CT010467.1
</td>
<td style="text-align:left;">
-3.433
</td>
<td style="text-align:left;">
10.638
</td>
<td style="text-align:left;">
-4.709
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
3.873
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm11168
</td>
<td style="text-align:left;">
-2.787
</td>
<td style="text-align:left;">
5.727
</td>
<td style="text-align:left;">
-4.545
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
3.231
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10722
</td>
<td style="text-align:left;">
-2.870
</td>
<td style="text-align:left;">
5.562
</td>
<td style="text-align:left;">
-4.538
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
3.202
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10800
</td>
<td style="text-align:left;">
-3.581
</td>
<td style="text-align:left;">
10.504
</td>
<td style="text-align:left;">
-4.509
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
3.091
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10717
</td>
<td style="text-align:left;">
-2.839
</td>
<td style="text-align:left;">
6.955
</td>
<td style="text-align:left;">
-4.471
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
2.948
</td>
</tr>
<tr>
<td style="text-align:left;">
Traf2
</td>
<td style="text-align:left;">
2.132
</td>
<td style="text-align:left;">
0.995
</td>
<td style="text-align:left;">
4.467
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
2.929
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm37144
</td>
<td style="text-align:left;">
1.894
</td>
<td style="text-align:left;">
1.609
</td>
<td style="text-align:left;">
4.456
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
2.888
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10801
</td>
<td style="text-align:left;">
-3.026
</td>
<td style="text-align:left;">
7.744
</td>
<td style="text-align:left;">
-4.421
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
2.758
</td>
</tr>
<tr>
<td style="text-align:left;">
Soat1
</td>
<td style="text-align:left;">
-3.263
</td>
<td style="text-align:left;">
2.004
</td>
<td style="text-align:left;">
-4.355
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.008
</td>
<td style="text-align:left;">
2.510
</td>
</tr>
<tr>
<td style="text-align:left;">
Cst7
</td>
<td style="text-align:left;">
-5.555
</td>
<td style="text-align:left;">
3.339
</td>
<td style="text-align:left;">
-4.335
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
2.434
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm20425
</td>
<td style="text-align:left;">
1.482
</td>
<td style="text-align:left;">
0.202
</td>
<td style="text-align:left;">
4.298
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.296
</td>
</tr>
<tr>
<td style="text-align:left;">
Slc25a5
</td>
<td style="text-align:left;">
-4.348
</td>
<td style="text-align:left;">
4.106
</td>
<td style="text-align:left;">
-4.297
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.293
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm15500
</td>
<td style="text-align:left;">
-4.193
</td>
<td style="text-align:left;">
4.285
</td>
<td style="text-align:left;">
-4.281
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.235
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm14303
</td>
<td style="text-align:left;">
-2.479
</td>
<td style="text-align:left;">
2.264
</td>
<td style="text-align:left;">
-4.279
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.229
</td>
</tr>
<tr>
<td style="text-align:left;">
Taf5l
</td>
<td style="text-align:left;">
1.117
</td>
<td style="text-align:left;">
0.123
</td>
<td style="text-align:left;">
4.261
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.162
</td>
</tr>
<tr>
<td style="text-align:left;">
Lipe
</td>
<td style="text-align:left;">
1.932
</td>
<td style="text-align:left;">
1.789
</td>
<td style="text-align:left;">
4.255
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
2.140
</td>
</tr>
<tr>
<td style="text-align:left;">
AW822252
</td>
<td style="text-align:left;">
1.266
</td>
<td style="text-align:left;">
0.732
</td>
<td style="text-align:left;">
4.211
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.012
</td>
<td style="text-align:left;">
1.980
</td>
</tr>
<tr>
<td style="text-align:left;">
Nckap5l
</td>
<td style="text-align:left;">
2.140
</td>
<td style="text-align:left;">
0.605
</td>
<td style="text-align:left;">
4.210
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.012
</td>
<td style="text-align:left;">
1.974
</td>
</tr>
<tr>
<td style="text-align:left;">
Rbm15
</td>
<td style="text-align:left;">
2.276
</td>
<td style="text-align:left;">
0.888
</td>
<td style="text-align:left;">
4.191
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.012
</td>
<td style="text-align:left;">
1.905
</td>
</tr>
<tr>
<td style="text-align:left;">
Ssr4
</td>
<td style="text-align:left;">
-3.766
</td>
<td style="text-align:left;">
2.953
</td>
<td style="text-align:left;">
-4.168
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
1.823
</td>
</tr>
<tr>
<td style="text-align:left;">
Cyba
</td>
<td style="text-align:left;">
-3.637
</td>
<td style="text-align:left;">
5.245
</td>
<td style="text-align:left;">
-4.157
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.014
</td>
<td style="text-align:left;">
1.783
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm43652
</td>
<td style="text-align:left;">
1.244
</td>
<td style="text-align:left;">
0.726
</td>
<td style="text-align:left;">
4.102
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.017
</td>
<td style="text-align:left;">
1.587
</td>
</tr>
<tr>
<td style="text-align:left;">
Clec7a
</td>
<td style="text-align:left;">
-4.460
</td>
<td style="text-align:left;">
2.762
</td>
<td style="text-align:left;">
-4.098
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.017
</td>
<td style="text-align:left;">
1.573
</td>
</tr>
<tr>
<td style="text-align:left;">
Fmo1
</td>
<td style="text-align:left;">
1.488
</td>
<td style="text-align:left;">
0.512
</td>
<td style="text-align:left;">
4.060
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.019
</td>
<td style="text-align:left;">
1.439
</td>
</tr>
<tr>
<td style="text-align:left;">
Fcer1g
</td>
<td style="text-align:left;">
-3.501
</td>
<td style="text-align:left;">
6.858
</td>
<td style="text-align:left;">
-4.038
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.020
</td>
<td style="text-align:left;">
1.362
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm45220
</td>
<td style="text-align:left;">
2.414
</td>
<td style="text-align:left;">
0.695
</td>
<td style="text-align:left;">
4.036
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.020
</td>
<td style="text-align:left;">
1.354
</td>
</tr>
<tr>
<td style="text-align:left;">
Tmem106c
</td>
<td style="text-align:left;">
2.262
</td>
<td style="text-align:left;">
0.711
</td>
<td style="text-align:left;">
4.012
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.022
</td>
<td style="text-align:left;">
1.269
</td>
</tr>
<tr>
<td style="text-align:left;">
Axl
</td>
<td style="text-align:left;">
-4.043
</td>
<td style="text-align:left;">
2.164
</td>
<td style="text-align:left;">
-4.009
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.022
</td>
<td style="text-align:left;">
1.258
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10443
</td>
<td style="text-align:left;">
-2.528
</td>
<td style="text-align:left;">
2.200
</td>
<td style="text-align:left;">
-4.003
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.022
</td>
<td style="text-align:left;">
1.238
</td>
</tr>
<tr>
<td style="text-align:left;">
Polr3gl
</td>
<td style="text-align:left;">
2.100
</td>
<td style="text-align:left;">
0.726
</td>
<td style="text-align:left;">
3.972
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.024
</td>
<td style="text-align:left;">
1.132
</td>
</tr>
<tr>
<td style="text-align:left;">
Rpl13a
</td>
<td style="text-align:left;">
-3.451
</td>
<td style="text-align:left;">
4.835
</td>
<td style="text-align:left;">
-3.960
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.025
</td>
<td style="text-align:left;">
1.090
</td>
</tr>
<tr>
<td style="text-align:left;">
Ifi27l2a
</td>
<td style="text-align:left;">
-3.174
</td>
<td style="text-align:left;">
1.810
</td>
<td style="text-align:left;">
-3.950
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.025
</td>
<td style="text-align:left;">
1.056
</td>
</tr>
<tr>
<td style="text-align:left;">
B2m
</td>
<td style="text-align:left;">
-3.469
</td>
<td style="text-align:left;">
7.942
</td>
<td style="text-align:left;">
-3.883
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.032
</td>
<td style="text-align:left;">
0.826
</td>
</tr>
<tr>
<td style="text-align:left;">
Rps8
</td>
<td style="text-align:left;">
-3.139
</td>
<td style="text-align:left;">
3.429
</td>
<td style="text-align:left;">
-3.877
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.032
</td>
<td style="text-align:left;">
0.807
</td>
</tr>
<tr>
<td style="text-align:left;">
Mzt2
</td>
<td style="text-align:left;">
1.955
</td>
<td style="text-align:left;">
0.682
</td>
<td style="text-align:left;">
3.796
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.043
</td>
<td style="text-align:left;">
0.533
</td>
</tr>
<tr>
<td style="text-align:left;">
H2-K1
</td>
<td style="text-align:left;">
-4.266
</td>
<td style="text-align:left;">
5.315
</td>
<td style="text-align:left;">
-3.788
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.044
</td>
<td style="text-align:left;">
0.507
</td>
</tr>
<tr>
<td style="text-align:left;">
Dnajc3
</td>
<td style="text-align:left;">
-3.409
</td>
<td style="text-align:left;">
1.992
</td>
<td style="text-align:left;">
-3.780
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.044
</td>
<td style="text-align:left;">
0.479
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm44645
</td>
<td style="text-align:left;">
1.097
</td>
<td style="text-align:left;">
0.462
</td>
<td style="text-align:left;">
3.770
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.045
</td>
<td style="text-align:left;">
0.447
</td>
</tr>
<tr>
<td style="text-align:left;">
Lamp2
</td>
<td style="text-align:left;">
-3.692
</td>
<td style="text-align:left;">
4.178
</td>
<td style="text-align:left;">
-3.767
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.045
</td>
<td style="text-align:left;">
0.436
</td>
</tr>
<tr>
<td style="text-align:left;">
Mtmr3
</td>
<td style="text-align:left;">
2.463
</td>
<td style="text-align:left;">
0.940
</td>
<td style="text-align:left;">
3.765
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.045
</td>
<td style="text-align:left;">
0.431
</td>
</tr>
<tr>
<td style="text-align:left;">
Tsix
</td>
<td style="text-align:left;">
1.075
</td>
<td style="text-align:left;">
0.243
</td>
<td style="text-align:left;">
3.753
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.046
</td>
<td style="text-align:left;">
0.392
</td>
</tr>
<tr>
<td style="text-align:left;">
Eng
</td>
<td style="text-align:left;">
2.294
</td>
<td style="text-align:left;">
0.862
</td>
<td style="text-align:left;">
3.742
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.047
</td>
<td style="text-align:left;">
0.355
</td>
</tr>
<tr>
<td style="text-align:left;">
Rps21
</td>
<td style="text-align:left;">
-2.554
</td>
<td style="text-align:left;">
2.191
</td>
<td style="text-align:left;">
-3.735
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.048
</td>
<td style="text-align:left;">
0.331
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>res_output[[&quot;plot&quot;]]</code></pre>
<div id="htmlwidget-be76d69923c41aaa24e1" style="width:1440px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-be76d69923c41aaa24e1">{"x":{"data":[{"x":[144.398379404524,141.273364669161,136.423671290723,143.121602379202,141.270790521932],"y":[4.30830901904371,2.196373085725,2.1662437192929,2.04410899703195,1.89366053220949],"text":["TF:M00285_1 (22) <br> Factor: TCF11; motif: GTCATNNWNNNNN; match class: 1 <br> 4.917e-05","TF:M00935_1 (10) <br> Factor: NFAT; motif: NWGGAAANWN; match class: 1 <br> 6.362e-03","TF:M00621_1 (164) <br> Factor: C/EBPdelta; motif: MATTKCNTMAYY; match class: 1 <br> 6.820e-03","TF:M07466_1 (224) <br> Factor: Six-1; motif: CTCARRTTWCN; match class: 1 <br> 9.034e-03","TF:M00935 (86) <br> Factor: NFAT; motif: NWGGAAANWN <br> 1.277e-02"],"key":["TF:M00285_1","TF:M00935_1","TF:M00621_1","TF:M07466_1","TF:M00935"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.8,"size":[13.2961317767835,3.77952755905512,21.7046023273465,22.6771653543307,19.5008849099126],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedData7bffe5d3","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,33.6673382117837],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[36.2408724020952,47.7419966985976],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[50.3155308889091,128.772294214747],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[131.345828405059,132.75869867554],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[135.332232865851,146.13850293097],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.712037121281,151.18520347817],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[153.758737668482,155.506167383704],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[158.079701574015,200.00257353419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":68.4975102650136,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"tickvals":[17.8336691058918,41.9914345503464,89.5439125518281,132.052263540299,140.73536789841,149.948620299726,154.632452526093,179.041137554103],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.3931091739311},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"192455e06ae4":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1924585a1bb2":{"x":{},"xend":{},"y":{},"yend":{}},"1924129150b8":{"x":{},"xend":{},"y":{},"yend":{}},"192470516a60":{"x":{},"xend":{},"y":{},"yend":{}},"19241f1c460d":{"x":{},"xend":{},"y":{},"yend":{}},"192455ca125b":{"x":{},"xend":{},"y":{},"yend":{}},"19242a932aa4":{"x":{},"xend":{},"y":{},"yend":{}},"192472065b7e":{"x":{},"xend":{},"y":{},"yend":{}},"192436eb4046":{"x":{},"xend":{},"y":{},"yend":{}},"19241a3329fe":{"x":{},"y":{}},"1924235b41fb":{"yintercept":{}}},"cur_data":"192455e06ae4","visdat":{"192455e06ae4":["function (y) ","x"],"1924585a1bb2":["function (y) ","x"],"1924129150b8":["function (y) ","x"],"192470516a60":["function (y) ","x"],"19241f1c460d":["function (y) ","x"],"192455ca125b":["function (y) ","x"],"19242a932aa4":["function (y) ","x"],"192472065b7e":["function (y) ","x"],"192436eb4046":["function (y) ","x"],"19241a3329fe":["function (y) ","x"],"1924235b41fb":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData7bffe5d3"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>get results table <a href="LMER_Microglia_0.xlsx">here</a></p>
</div>
<div id="microglia_1" class="section level3">
<h3>Microglia_1</h3>
<pre class="r"><code>res_output=generate_output(&quot;Microglia_1&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/Microglia_1-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
logFC
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
AveExpr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
t
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
P.Value
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
adj.P.Val
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rn7s1
</td>
<td style="text-align:left;">
5.171
</td>
<td style="text-align:left;">
2.251
</td>
<td style="text-align:left;">
15.706
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
55.566
</td>
</tr>
<tr>
<td style="text-align:left;">
AY036118
</td>
<td style="text-align:left;">
-7.747
</td>
<td style="text-align:left;">
4.063
</td>
<td style="text-align:left;">
-8.414
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
19.581
</td>
</tr>
<tr>
<td style="text-align:left;">
Xist
</td>
<td style="text-align:left;">
4.480
</td>
<td style="text-align:left;">
0.834
</td>
<td style="text-align:left;">
8.298
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
19.011
</td>
</tr>
<tr>
<td style="text-align:left;">
Pola2
</td>
<td style="text-align:left;">
-3.212
</td>
<td style="text-align:left;">
1.551
</td>
<td style="text-align:left;">
-7.042
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
13.071
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s2
</td>
<td style="text-align:left;">
3.739
</td>
<td style="text-align:left;">
1.658
</td>
<td style="text-align:left;">
6.835
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
12.132
</td>
</tr>
<tr>
<td style="text-align:left;">
AC099934.1
</td>
<td style="text-align:left;">
1.435
</td>
<td style="text-align:left;">
0.595
</td>
<td style="text-align:left;">
6.407
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
10.233
</td>
</tr>
<tr>
<td style="text-align:left;">
Mgmt
</td>
<td style="text-align:left;">
2.314
</td>
<td style="text-align:left;">
1.070
</td>
<td style="text-align:left;">
5.923
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
8.166
</td>
</tr>
<tr>
<td style="text-align:left;">
Tsix
</td>
<td style="text-align:left;">
2.028
</td>
<td style="text-align:left;">
0.342
</td>
<td style="text-align:left;">
5.846
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
7.844
</td>
</tr>
<tr>
<td style="text-align:left;">
Tbata
</td>
<td style="text-align:left;">
1.984
</td>
<td style="text-align:left;">
1.018
</td>
<td style="text-align:left;">
4.947
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
4.306
</td>
</tr>
<tr>
<td style="text-align:left;">
Tmem170
</td>
<td style="text-align:left;">
2.015
</td>
<td style="text-align:left;">
1.087
</td>
<td style="text-align:left;">
4.917
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
4.196
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm26870
</td>
<td style="text-align:left;">
-4.250
</td>
<td style="text-align:left;">
2.611
</td>
<td style="text-align:left;">
-4.821
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
3.845
</td>
</tr>
<tr>
<td style="text-align:left;">
Zfp992
</td>
<td style="text-align:left;">
-2.797
</td>
<td style="text-align:left;">
0.847
</td>
<td style="text-align:left;">
-4.604
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
3.065
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10719
</td>
<td style="text-align:left;">
-3.041
</td>
<td style="text-align:left;">
1.702
</td>
<td style="text-align:left;">
-4.520
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.012
</td>
<td style="text-align:left;">
2.771
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm15564
</td>
<td style="text-align:left;">
1.545
</td>
<td style="text-align:left;">
1.085
</td>
<td style="text-align:left;">
4.493
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
2.678
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10801
</td>
<td style="text-align:left;">
-3.336
</td>
<td style="text-align:left;">
1.981
</td>
<td style="text-align:left;">
-4.468
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
2.591
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10800
</td>
<td style="text-align:left;">
-4.381
</td>
<td style="text-align:left;">
3.697
</td>
<td style="text-align:left;">
-4.258
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.030
</td>
<td style="text-align:left;">
1.878
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10722
</td>
<td style="text-align:left;">
-2.146
</td>
<td style="text-align:left;">
0.803
</td>
<td style="text-align:left;">
-4.174
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.039
</td>
<td style="text-align:left;">
1.603
</td>
</tr>
<tr>
<td style="text-align:left;">
Cst7
</td>
<td style="text-align:left;">
-4.417
</td>
<td style="text-align:left;">
1.822
</td>
<td style="text-align:left;">
-4.162
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.039
</td>
<td style="text-align:left;">
1.563
</td>
</tr>
<tr>
<td style="text-align:left;">
Slc6a20b
</td>
<td style="text-align:left;">
1.468
</td>
<td style="text-align:left;">
0.968
</td>
<td style="text-align:left;">
4.101
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.046
</td>
<td style="text-align:left;">
1.366
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm26822
</td>
<td style="text-align:left;">
1.544
</td>
<td style="text-align:left;">
1.578
</td>
<td style="text-align:left;">
4.092
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.046
</td>
<td style="text-align:left;">
1.336
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>res_output[[&quot;plot&quot;]]</code></pre>
<div id="htmlwidget-c86bfa0412dc20f269ee" style="width:1440px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-c86bfa0412dc20f269ee">{"x":{"data":[{"x":[144.398379404524],"y":[3.02318423097031],"text":"TF:M00285_1 (22) <br> Factor: TCF11; motif: GTCATNNWNNNNN; match class: 1 <br> 9.480e-04","key":["TF:M00285_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.8,"size":17.1421753925017,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedDatabe76d699","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[2,33.6673382117837],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[36.2408724020952,47.7419966985976],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[50.3155308889091,128.772294214747],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[131.345828405059,132.75869867554],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[135.332232865851,146.13850293097],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.712037121281,151.18520347817],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[153.758737668482,155.506167383704],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[158.079701574015,200.00257353419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":68.4975102650136,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"tickvals":[17.8336691058918,41.9914345503464,89.5439125518281,132.052263540299,140.73536789841,149.948620299726,154.632452526093,179.041137554103],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.3931091739311},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"19242d36a04":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"19246063413f":{"x":{},"xend":{},"y":{},"yend":{}},"19241dbe54f3":{"x":{},"xend":{},"y":{},"yend":{}},"19247e3a3a9a":{"x":{},"xend":{},"y":{},"yend":{}},"19245b6c24f3":{"x":{},"xend":{},"y":{},"yend":{}},"1924546979c3":{"x":{},"xend":{},"y":{},"yend":{}},"19245fc972a7":{"x":{},"xend":{},"y":{},"yend":{}},"19241ee1984":{"x":{},"xend":{},"y":{},"yend":{}},"19243e03550":{"x":{},"xend":{},"y":{},"yend":{}},"192429a9624a":{"x":{},"y":{}},"192473a128e3":{"yintercept":{}}},"cur_data":"19242d36a04","visdat":{"19242d36a04":["function (y) ","x"],"19246063413f":["function (y) ","x"],"19241dbe54f3":["function (y) ","x"],"19247e3a3a9a":["function (y) ","x"],"19245b6c24f3":["function (y) ","x"],"1924546979c3":["function (y) ","x"],"19245fc972a7":["function (y) ","x"],"19241ee1984":["function (y) ","x"],"19243e03550":["function (y) ","x"],"192429a9624a":["function (y) ","x"],"192473a128e3":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDatabe76d699"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>get results table <a href="LMER_Microglia_1.xlsx">here</a></p>
</div>
<div id="microglia_2" class="section level3">
<h3>Microglia_2</h3>
<pre class="r"><code>res_output=generate_output(&quot;Microglia_2&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/Microglia_2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
logFC
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
AveExpr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
t
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
P.Value
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
adj.P.Val
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AY036118
</td>
<td style="text-align:left;">
-10.106
</td>
<td style="text-align:left;">
5.877
</td>
<td style="text-align:left;">
-13.790
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
40.529
</td>
</tr>
<tr>
<td style="text-align:left;">
AC099934.1
</td>
<td style="text-align:left;">
2.240
</td>
<td style="text-align:left;">
0.649
</td>
<td style="text-align:left;">
13.121
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
37.942
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s1
</td>
<td style="text-align:left;">
5.881
</td>
<td style="text-align:left;">
2.015
</td>
<td style="text-align:left;">
12.703
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
36.288
</td>
</tr>
<tr>
<td style="text-align:left;">
Tbata
</td>
<td style="text-align:left;">
3.400
</td>
<td style="text-align:left;">
1.094
</td>
<td style="text-align:left;">
9.836
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
24.360
</td>
</tr>
<tr>
<td style="text-align:left;">
Tmem170
</td>
<td style="text-align:left;">
2.548
</td>
<td style="text-align:left;">
0.931
</td>
<td style="text-align:left;">
8.343
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
17.927
</td>
</tr>
<tr>
<td style="text-align:left;">
Mgmt
</td>
<td style="text-align:left;">
3.287
</td>
<td style="text-align:left;">
1.015
</td>
<td style="text-align:left;">
8.171
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
17.183
</td>
</tr>
<tr>
<td style="text-align:left;">
Gab3
</td>
<td style="text-align:left;">
4.652
</td>
<td style="text-align:left;">
1.369
</td>
<td style="text-align:left;">
7.325
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
13.580
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10719
</td>
<td style="text-align:left;">
-5.301
</td>
<td style="text-align:left;">
4.443
</td>
<td style="text-align:left;">
-7.065
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
12.488
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm26870
</td>
<td style="text-align:left;">
-6.685
</td>
<td style="text-align:left;">
6.156
</td>
<td style="text-align:left;">
-6.989
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
12.169
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10800
</td>
<td style="text-align:left;">
-6.887
</td>
<td style="text-align:left;">
7.447
</td>
<td style="text-align:left;">
-6.841
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
11.553
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm15564
</td>
<td style="text-align:left;">
2.706
</td>
<td style="text-align:left;">
1.222
</td>
<td style="text-align:left;">
6.700
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
10.972
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10801
</td>
<td style="text-align:left;">
-5.372
</td>
<td style="text-align:left;">
5.022
</td>
<td style="text-align:left;">
-6.609
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
10.599
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10717
</td>
<td style="text-align:left;">
-4.477
</td>
<td style="text-align:left;">
4.416
</td>
<td style="text-align:left;">
-6.084
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
8.477
</td>
</tr>
<tr>
<td style="text-align:left;">
Prrx1
</td>
<td style="text-align:left;">
1.709
</td>
<td style="text-align:left;">
0.589
</td>
<td style="text-align:left;">
5.548
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
6.389
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s2
</td>
<td style="text-align:left;">
3.611
</td>
<td style="text-align:left;">
1.358
</td>
<td style="text-align:left;">
5.396
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
5.817
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm37716
</td>
<td style="text-align:left;">
1.631
</td>
<td style="text-align:left;">
0.489
</td>
<td style="text-align:left;">
4.840
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
3.786
</td>
</tr>
<tr>
<td style="text-align:left;">
Xist
</td>
<td style="text-align:left;">
3.446
</td>
<td style="text-align:left;">
0.711
</td>
<td style="text-align:left;">
4.830
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
3.750
</td>
</tr>
<tr>
<td style="text-align:left;">
Pola2
</td>
<td style="text-align:left;">
-2.603
</td>
<td style="text-align:left;">
2.295
</td>
<td style="text-align:left;">
-4.758
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
3.499
</td>
</tr>
<tr>
<td style="text-align:left;">
CT010467.1
</td>
<td style="text-align:left;">
-1.778
</td>
<td style="text-align:left;">
11.024
</td>
<td style="text-align:left;">
-4.746
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
3.456
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm11168
</td>
<td style="text-align:left;">
-3.190
</td>
<td style="text-align:left;">
3.320
</td>
<td style="text-align:left;">
-4.726
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
3.386
</td>
</tr>
<tr>
<td style="text-align:left;">
A430010J10Rik
</td>
<td style="text-align:left;">
1.554
</td>
<td style="text-align:left;">
1.105
</td>
<td style="text-align:left;">
4.427
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.015
</td>
<td style="text-align:left;">
2.363
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm44068
</td>
<td style="text-align:left;">
-1.496
</td>
<td style="text-align:left;">
0.544
</td>
<td style="text-align:left;">
-4.406
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.016
</td>
<td style="text-align:left;">
2.294
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm28661
</td>
<td style="text-align:left;">
-5.071
</td>
<td style="text-align:left;">
4.633
</td>
<td style="text-align:left;">
-4.295
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.023
</td>
<td style="text-align:left;">
1.928
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm22567
</td>
<td style="text-align:left;">
1.380
</td>
<td style="text-align:left;">
0.525
</td>
<td style="text-align:left;">
4.222
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.029
</td>
<td style="text-align:left;">
1.689
</td>
</tr>
<tr>
<td style="text-align:left;">
AC127302.1
</td>
<td style="text-align:left;">
-2.557
</td>
<td style="text-align:left;">
3.170
</td>
<td style="text-align:left;">
-4.213
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.029
</td>
<td style="text-align:left;">
1.659
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm21738
</td>
<td style="text-align:left;">
-3.628
</td>
<td style="text-align:left;">
3.300
</td>
<td style="text-align:left;">
-4.176
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.032
</td>
<td style="text-align:left;">
1.539
</td>
</tr>
<tr>
<td style="text-align:left;">
Rpl19-ps11
</td>
<td style="text-align:left;">
-1.655
</td>
<td style="text-align:left;">
0.404
</td>
<td style="text-align:left;">
-4.115
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.039
</td>
<td style="text-align:left;">
1.345
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm10359
</td>
<td style="text-align:left;">
2.833
</td>
<td style="text-align:left;">
1.183
</td>
<td style="text-align:left;">
4.036
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.049
</td>
<td style="text-align:left;">
1.094
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm12671
</td>
<td style="text-align:left;">
2.833
</td>
<td style="text-align:left;">
1.183
</td>
<td style="text-align:left;">
4.036
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.049
</td>
<td style="text-align:left;">
1.094
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>res_output[[&quot;plot&quot;]]</code></pre>
<div id="htmlwidget-709fe70a790a38ae983a" style="width:1440px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-709fe70a790a38ae983a">{"x":{"data":[{"x":[56.9812030888081],"y":[2.25141804601667],"text":"GO:0006617 (6) <br> SRP-dependent cotranslational protein targeting to membrane, signal sequence recognition <br> 5.605e-03","key":["GO:0006617"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":3.77952755905512,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData23c41aaa","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[144.398379404524,141.270790521932,141.273364669161],"y":[4.73877443652452,1.65151829365121,1.47538526269093],"text":["TF:M00285_1 (22) <br> Factor: TCF11; motif: GTCATNNWNNNNN; match class: 1 <br> 1.825e-05","TF:M00935 (86) <br> Factor: NFAT; motif: NWGGAAANWN <br> 2.231e-02","TF:M00935_1 (10) <br> Factor: NFAT; motif: NWGGAAANWN; match class: 1 <br> 3.347e-02"],"key":["TF:M00285_1","TF:M00935","TF:M00935_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.8,"size":[16.9805456155825,22.6771653543307,12.0568883724757],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedData23c41aaa","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,33.6673382117837],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[36.2408724020952,47.7419966985976],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[50.3155308889091,128.772294214747],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[131.345828405059,132.75869867554],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[135.332232865851,146.13850293097],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.712037121281,151.18520347817],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[153.758737668482,155.506167383704],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[158.079701574015,200.00257353419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":68.4975102650136,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"tickvals":[17.8336691058918,41.9914345503464,89.5439125518281,132.052263540299,140.73536789841,149.948620299726,154.632452526093,179.041137554103],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","TF","MIRNA","CORUM","HP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.3931091739311},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"1924ba0522b":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"192436acf7a":{"x":{},"xend":{},"y":{},"yend":{}},"19242db21e81":{"x":{},"xend":{},"y":{},"yend":{}},"192453954f43":{"x":{},"xend":{},"y":{},"yend":{}},"19246f6f483a":{"x":{},"xend":{},"y":{},"yend":{}},"1924563ac96":{"x":{},"xend":{},"y":{},"yend":{}},"1924230849b4":{"x":{},"xend":{},"y":{},"yend":{}},"192479e84cf4":{"x":{},"xend":{},"y":{},"yend":{}},"192472da4d96":{"x":{},"xend":{},"y":{},"yend":{}},"192475855254":{"x":{},"y":{}},"1924539278c1":{"yintercept":{}}},"cur_data":"1924ba0522b","visdat":{"1924ba0522b":["function (y) ","x"],"192436acf7a":["function (y) ","x"],"19242db21e81":["function (y) ","x"],"192453954f43":["function (y) ","x"],"19246f6f483a":["function (y) ","x"],"1924563ac96":["function (y) ","x"],"1924230849b4":["function (y) ","x"],"192479e84cf4":["function (y) ","x"],"192472da4d96":["function (y) ","x"],"192475855254":["function (y) ","x"],"1924539278c1":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData23c41aaa"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>get results table <a href="LMER_Microglia_2.xlsx">here</a></p>
</div>
<div id="microglia_3" class="section level3">
<h3>Microglia_3</h3>
<pre class="r"><code>res_output=generate_output(&quot;Microglia_3&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/Microglia_3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;no significant GO terms identified&quot;</code></pre>
<pre class="r"><code>res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
logFC
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
AveExpr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
t
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
P.Value
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
adj.P.Val
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Apoe
</td>
<td style="text-align:left;">
-10.329
</td>
<td style="text-align:left;">
8.936
</td>
<td style="text-align:left;">
-11.161
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
27.508
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s1
</td>
<td style="text-align:left;">
4.871
</td>
<td style="text-align:left;">
1.890
</td>
<td style="text-align:left;">
10.710
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
25.819
</td>
</tr>
<tr>
<td style="text-align:left;">
Tmem170
</td>
<td style="text-align:left;">
2.313
</td>
<td style="text-align:left;">
0.742
</td>
<td style="text-align:left;">
9.944
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
22.898
</td>
</tr>
<tr>
<td style="text-align:left;">
AY036118
</td>
<td style="text-align:left;">
-8.824
</td>
<td style="text-align:left;">
5.396
</td>
<td style="text-align:left;">
-6.850
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
10.815
</td>
</tr>
<tr>
<td style="text-align:left;">
Tbata
</td>
<td style="text-align:left;">
2.553
</td>
<td style="text-align:left;">
0.831
</td>
<td style="text-align:left;">
6.560
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
9.704
</td>
</tr>
<tr>
<td style="text-align:left;">
Cst7
</td>
<td style="text-align:left;">
-6.295
</td>
<td style="text-align:left;">
5.374
</td>
<td style="text-align:left;">
-5.748
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
6.680
</td>
</tr>
<tr>
<td style="text-align:left;">
Rn7s2
</td>
<td style="text-align:left;">
3.502
</td>
<td style="text-align:left;">
1.489
</td>
<td style="text-align:left;">
5.388
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
5.390
</td>
</tr>
<tr>
<td style="text-align:left;">
Mgmt
</td>
<td style="text-align:left;">
2.551
</td>
<td style="text-align:left;">
0.931
</td>
<td style="text-align:left;">
5.248
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
4.901
</td>
</tr>
<tr>
<td style="text-align:left;">
Gm15564
</td>
<td style="text-align:left;">
2.059
</td>
<td style="text-align:left;">
1.137
</td>
<td style="text-align:left;">
5.108
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
4.415
</td>
</tr>
<tr>
<td style="text-align:left;">
Lyz2
</td>
<td style="text-align:left;">
-4.371
</td>
<td style="text-align:left;">
7.997
</td>
<td style="text-align:left;">
-5.064
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
4.264
</td>
</tr>
<tr>
<td style="text-align:left;">
Xist
</td>
<td style="text-align:left;">
3.215
</td>
<td style="text-align:left;">
0.580
</td>
<td style="text-align:left;">
5.053
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
4.228
</td>
</tr>
<tr>
<td style="text-align:left;">
AC099934.1
</td>
<td style="text-align:left;">
1.361
</td>
<td style="text-align:left;">
0.399
</td>
<td style="text-align:left;">
4.751
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.008
</td>
<td style="text-align:left;">
3.211
</td>
</tr>
<tr>
<td style="text-align:left;">
Ctsb
</td>
<td style="text-align:left;">
-2.050
</td>
<td style="text-align:left;">
9.953
</td>
<td style="text-align:left;">
-4.472
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.022
</td>
<td style="text-align:left;">
2.301
</td>
</tr>
<tr>
<td style="text-align:left;">
Prrx1
</td>
<td style="text-align:left;">
1.217
</td>
<td style="text-align:left;">
0.338
</td>
<td style="text-align:left;">
4.398
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.027
</td>
<td style="text-align:left;">
2.068
</td>
</tr>
<tr>
<td style="text-align:left;">
Clec7a
</td>
<td style="text-align:left;">
-5.414
</td>
<td style="text-align:left;">
3.980
</td>
<td style="text-align:left;">
-4.293
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.038
</td>
<td style="text-align:left;">
1.736
</td>
</tr>
<tr>
<td style="text-align:left;">
Slc25a37
</td>
<td style="text-align:left;">
3.907
</td>
<td style="text-align:left;">
1.361
</td>
<td style="text-align:left;">
4.258
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.041
</td>
<td style="text-align:left;">
1.628
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>res_output[[&quot;plot&quot;]]</code></pre>
<pre><code>[1] &quot;no significant enrichment identified&quot;</code></pre>
<p>get results table <a href="LMER_Microglia_3.xlsx">here</a></p>
</div>
<div id="microglia_4" class="section level3">
<h3>Microglia_4</h3>
<p>calculation not converging as Celltype not identified in all conditions however also rare in other conditions, no significant difference</p>
<pre class="r"><code>idx=samplemeta$Celltype==&quot;Microglia_4&quot;
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])</code></pre>
<pre><code>        
         WT APPPS1+
  Ctrl   11      10
  Stroke  4       1</code></pre>
<pre class="r"><code>table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %&gt;% fisher.test()</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  .
p-value = 0.3562
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.005090941 3.588141911
sample estimates:
odds ratio 
 0.2877466 </code></pre>
<pre class="r"><code>#res_output=generate_output(&quot;Microglia_4&quot;)
#res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()
#res_output[[&quot;plot&quot;]]</code></pre>
<p>get results table <a href="LMER_Microglia_4.xlsx">here</a></p>
</div>
<div id="microglia_5" class="section level3">
<h3>Microglia_5</h3>
<pre class="r"><code>res_output=generate_output(&quot;Microglia_5&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/Microglia_5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;no significant GO terms identified&quot;</code></pre>
<pre class="r"><code>res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
logFC
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
AveExpr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
t
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
P.Value
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
adj.P.Val
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Strip1
</td>
<td style="text-align:left;">
5.337
</td>
<td style="text-align:left;">
0.696
</td>
<td style="text-align:left;">
15.181
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
10.786
</td>
</tr>
<tr>
<td style="text-align:left;">
Naalad2
</td>
<td style="text-align:left;">
6.931
</td>
<td style="text-align:left;">
0.908
</td>
<td style="text-align:left;">
12.659
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
9.531
</td>
</tr>
<tr>
<td style="text-align:left;">
Ino80
</td>
<td style="text-align:left;">
5.064
</td>
<td style="text-align:left;">
0.811
</td>
<td style="text-align:left;">
9.524
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
7.170
</td>
</tr>
<tr>
<td style="text-align:left;">
AC131743.1
</td>
<td style="text-align:left;">
1.546
</td>
<td style="text-align:left;">
0.200
</td>
<td style="text-align:left;">
7.232
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
4.604
</td>
</tr>
<tr>
<td style="text-align:left;">
Slc2a3
</td>
<td style="text-align:left;">
1.734
</td>
<td style="text-align:left;">
0.289
</td>
<td style="text-align:left;">
6.052
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0.019
</td>
<td style="text-align:left;">
2.928
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>res_output[[&quot;plot&quot;]]</code></pre>
<pre><code>[1] &quot;no significant enrichment identified&quot;</code></pre>
<p>get results table <a href="LMER_Microglia_5.xlsx">here</a></p>
</div>
<div id="granulozytes" class="section level3">
<h3>Granulozytes</h3>
<p>calculation not converging as Celltype not identified in all conditions however also rare in other conditions, no significant difference</p>
<pre class="r"><code>idx=samplemeta$Celltype==&quot;Granulocytes&quot;
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])</code></pre>
<pre><code>        
         WT APPPS1+
  Ctrl    0       2
  Stroke  6      14</code></pre>
<pre class="r"><code>table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %&gt;% fisher.test()</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  .
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.0000 14.7518
sample estimates:
odds ratio 
         0 </code></pre>
<pre class="r"><code># res_output=generate_output(&quot;Granulocytes &quot;)
# res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()
# res_output[[&quot;plot&quot;]]</code></pre>
<p>get results table <a href="LMER_Granulozytes.xlsx">here</a></p>
</div>
<div id="t_nk" class="section level3">
<h3>T_NK</h3>
<pre class="r"><code>idx=samplemeta$Celltype==&quot;T/NK&quot;
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])</code></pre>
<pre><code>        
         WT APPPS1+
  Ctrl    1       2
  Stroke  2      17</code></pre>
<pre class="r"><code>table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %&gt;% fisher.test()</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  .
p-value = 0.3708
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0481749 117.2316789
sample estimates:
odds ratio 
  3.869104 </code></pre>
<pre class="r"><code># res_output=generate_output(&quot;T/NK&quot;)
# res_output[[&quot;results_sig&quot;]] %&gt;% display_tab()
# res_output[[&quot;plot&quot;]]</code></pre>
<p>get results table <a href="LMER_T_NK.xlsx">here</a></p>
</div>
</div>
<div id="gene-expression-plot" class="section level2">
<h2>Gene expression plot</h2>
<iframe src="https://molgenlab.shinyapps.io/geneviewer/" height="900px" width="900px">
</iframe>
<button onclick="window.location.href=&#39;https://molgenlab.shinyapps.io/geneviewer/&#39;">
Open in Browser
</button>
<p>
</p>
<p>
</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                    LC_TIME=German_Germany.1252    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gprofiler2_0.2.1            lm.beta_1.5-1               pheatmap_1.0.12             RColorBrewer_1.1-2          kableExtra_1.3.4            viridis_0.6.2               viridisLite_0.4.0           xlsx_0.6.5                  brms_2.16.3                 Rcpp_1.0.8                  compareGroups_4.5.1         data.table_1.14.2           SingleR_1.8.1               SeuratObject_4.0.4          Seurat_4.1.0                forcats_0.5.1               stringr_1.4.0               dplyr_1.0.7                 purrr_0.3.4                 readr_2.1.2                 tidyr_1.1.4                 tibble_3.1.6                ggplot2_3.3.5               tidyverse_1.3.1             DESeq2_1.34.0               SummarizedExperiment_1.24.0 Biobase_2.54.0              MatrixGenerics_1.6.0        matrixStats_0.61.0          GenomicRanges_1.46.1        GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.3            BiocGenerics_0.40.0         limma_3.50.0                workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] scattermore_0.7           coda_0.19-4               bit64_4.0.5               knitr_1.37                dygraphs_1.1.1.6          irlba_2.3.5               DelayedArray_0.20.0       inline_0.3.19             rpart_4.1.16              KEGGREST_1.34.0           RCurl_1.98-1.5            generics_0.1.2            ScaledMatrix_1.2.0        callr_3.7.0               cowplot_1.1.1             RSQLite_2.2.9             mice_3.14.0               RANN_2.6.1                future_1.23.0             chron_2.3-56              bit_4.0.4                 tzdb_0.2.0                spatstat.data_2.1-2       webshot_0.5.2             xml2_1.3.3                lubridate_1.8.0           httpuv_1.6.5              StanHeaders_2.21.0-7      assertthat_0.2.1          xfun_0.29                 rJava_1.0-6               hms_1.1.1                 jquerylib_0.1.4           bayesplot_1.8.1           evaluate_0.14             promises_1.2.0.1          fansi_1.0.2               dbplyr_2.1.1              readxl_1.3.1              igraph_1.2.11             DBI_1.1.2                 geneplotter_1.72.0        Rsolnp_1.16               htmlwidgets_1.5.4         tensorA_0.36.2            spatstat.geom_2.3-1       ellipsis_0.3.2            crosstalk_1.2.0           backports_1.4.1           markdown_1.1              annotate_1.72.0           RcppParallel_5.1.5        deldir_1.0-6              sparseMatrixStats_1.6.0   vctrs_0.3.8               ROCR_1.0-11               abind_1.4-5               cachem_1.0.6              withr_2.4.3               packrat_0.7.0             HardyWeinberg_1.7.4       checkmate_2.0.0           sctransform_0.3.3         prettyunits_1.1.1         xts_0.12.1                goftest_1.2-3             svglite_2.0.0             cluster_2.1.2             lazyeval_0.2.2            crayon_1.4.2              genefilter_1.76.0         pkgconfig_2.0.3           nlme_3.1-155              nnet_7.3-17               rlang_1.0.0               globals_0.14.0            lifecycle_1.0.1           miniUI_0.1.1.1            colourpicker_1.1.1        modelr_0.1.8              rsvd_1.0.5                distributional_0.3.0      cellranger_1.1.0          rprojroot_2.0.2           polyclip_1.10-0           lmtest_0.9-39             flextable_0.6.10          Matrix_1.4-0              loo_2.4.1                 zoo_1.8-9                 reprex_2.0.1              base64enc_0.1-3           whisker_0.4               ggridges_0.5.3            processx_3.5.2            png_0.1-7                 bitops_1.0-7              getPass_0.2-2             KernSmooth_2.23-20        Biostrings_2.62.0         blob_1.2.2                DelayedMatrixStats_1.16.0 parallelly_1.30.0         shinystan_2.5.0           beachmat_2.10.0           scales_1.1.1              memoise_2.0.1             magrittr_2.0.2            plyr_1.8.6                ica_1.0-2                 threejs_0.3.3             zlibbioc_1.40.0           compiler_4.1.2            rstantools_2.1.1          fitdistrplus_1.1-6        cli_3.1.1                 XVector_0.34.0            listenv_0.8.0             Brobdingnag_1.2-7         patchwork_1.1.1           pbapply_1.5-0             ps_1.6.0                  MASS_7.3-55               mgcv_1.8-38               tidyselect_1.1.1          stringi_1.7.6             highr_0.9                 yaml_2.2.2                askpass_1.1               BiocSingular_1.10.0       locfit_1.5-9.4            ggrepel_0.9.1             bridgesampling_1.1-2      grid_4.1.2                sass_0.4.0                tools_4.1.2               future.apply_1.8.1        parallel_4.1.2            rstudioapi_0.13           uuid_1.0-3                git2r_0.29.0              gridExtra_2.3             farver_2.1.0              posterior_1.2.0           Rtsne_0.15                BiocManager_1.30.16       digest_0.6.29             shiny_1.7.1               broom_0.7.12              later_1.3.0               writexl_1.4.0             RcppAnnoy_0.0.19          httr_1.4.2                gdtools_0.2.3             AnnotationDbi_1.56.2      rsconnect_0.8.25          colorspace_2.0-2          rvest_1.0.2               XML_3.99-0.8              fs_1.5.2                  tensor_1.5                reticulate_1.24           truncnorm_1.0-8           splines_4.1.2             statmod_1.4.36            uwot_0.1.11               spatstat.utils_2.3-0      xlsxjars_0.6.1            shinythemes_1.2.0         plotly_4.10.0             systemfonts_1.0.3         xtable_1.8-4              jsonlite_1.7.3            rstan_2.21.3              R6_2.5.1                  pillar_1.7.0              htmltools_0.5.2           mime_0.12                 DT_0.20                   glue_1.6.1                fastmap_1.1.0             BiocParallel_1.28.3       BiocNeighbors_1.12.0      codetools_0.2-18          pkgbuild_1.3.1            mvtnorm_1.1-3             utf8_1.2.2                lattice_0.20-45           bslib_0.3.1               spatstat.sparse_2.1-0     curl_4.3.2                leiden_0.3.9              gtools_3.9.2              officer_0.4.1             openssl_1.4.6             shinyjs_2.1.0             zip_2.2.0                 survival_3.2-13           rmarkdown_2.11            munsell_0.5.0             GenomeInfoDbData_1.2.7    haven_2.4.3               reshape2_1.4.4            gtable_0.3.0              spatstat.core_2.3-2      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
