<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AGC" />


<title>Biostatistical analysis part 2</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scSeq_Hefendehl</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Biostatistical analysis part 2</h1>
<h4 class="author">AGC</h4>
<h4 class="date">2 2 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>S:/KJP_Biolabor/Projects/scSeq_Hefendehl/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220131code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220131)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220131code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220131)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltree64cd37cb31ab6bb8935fdf508d12232202b9de90targetblank64cd37ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/64cd37cb31ab6bb8935fdf508d12232202b9de90" target="_blank">64cd37c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltree64cd37cb31ab6bb8935fdf508d12232202b9de90targetblank64cd37ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/64cd37cb31ab6bb8935fdf508d12232202b9de90" target="_blank">64cd37c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/ReloadAllData_Hefendehl_Stroke_Dec'21.RData
    Ignored:    data/Sample_Tables/
    Ignored:    data/counts.csv
    Ignored:    data/genecounts.csv
    Ignored:    data/microglia_protein.rds
    Ignored:    data/samples.integrated.RData
    Ignored:    data/tx2genes.csv

Untracked files:
    Untracked:  workflow_helper.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#get sample data
samples.integrated@meta.data %&gt;% as.data.frame() -&gt; samplemeta

# convert to correct data type 

# define gentype as is
samplemeta$Genotype_corr = factor(samplemeta$Genotype==&quot;wt&quot;, levels=c(F,T), labels = c(&quot;APPPS1+&quot;, &quot;WT&quot;))
samplemeta$Genotype_corr = relevel(samplemeta$Genotype_corr, ref=&quot;WT&quot;)

samplemeta$methoxy = factor(samplemeta$Genotype==&quot;MX04+&quot;, levels=c(T,F), labels = c(&quot;MX04+&quot;, &quot;MX04-&quot;))

samplemeta$Treatment = as.factor(samplemeta$Treatment)
samplemeta$Treatment = relevel(samplemeta$Treatment, ref=&quot;Ctrl&quot;)
samplemeta$Mouse_ID = as.factor(samplemeta$Mouse_ID)
samplemeta$Sex = as.factor(samplemeta$Sex)
samplemeta$Brain_region = as.factor(samplemeta$Brain_region)
samplemeta$Celltype = as.factor(samplemeta$Celltype)

nCells=nrow(samplemeta)
nMice=nlevels(samplemeta$Mouse_ID)
nCelltypes=nlevels(samplemeta$Celltype)</code></pre>
<div id="sample-descriptive" class="section level2">
<h2>Sample descriptive:</h2>
<p>Data contains a total of 649 Cells from 9. <em>Q:</em> Original raw datset containing only frankfurt data included 1149 cells. What were the filter criteria in the primary cell type analysis</p>
<p><strong>Cells per Mouse</strong></p>
<pre class="r"><code>table(Mouse_ID=samplemeta$Mouse_ID) %&gt;% as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Mouse_ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
23#15773
</td>
<td style="text-align:left;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
23#15774
</td>
<td style="text-align:left;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
23#15792
</td>
<td style="text-align:left;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
386
</td>
<td style="text-align:left;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
387
</td>
<td style="text-align:left;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
388
</td>
<td style="text-align:left;">
113
</td>
</tr>
<tr>
<td style="text-align:left;">
409
</td>
<td style="text-align:left;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
457
</td>
<td style="text-align:left;">
115
</td>
</tr>
<tr>
<td style="text-align:left;">
461
</td>
<td style="text-align:left;">
88
</td>
</tr>
</tbody>
</table>
</div>
<p>Cells per Strain</p>
<pre class="r"><code>table(Strain=samplemeta$Genotype_corr, Treatment=samplemeta$Treatment) %&gt;% as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Strain
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Treatment
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Ctrl
</td>
<td style="text-align:left;">
162
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Ctrl
</td>
<td style="text-align:left;">
228
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Stroke
</td>
<td style="text-align:left;">
127
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Stroke
</td>
<td style="text-align:left;">
132
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>table(Strain=samplemeta$Genotype_corr, Celltype=samplemeta$Celltype) %&gt;% 
  as.data.frame() %&gt;% display_tab()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Strain
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Celltype
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
T/NK
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
T/NK
</td>
<td style="text-align:left;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_0
</td>
<td style="text-align:left;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_0
</td>
<td style="text-align:left;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_1
</td>
<td style="text-align:left;">
93
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_1
</td>
<td style="text-align:left;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_2
</td>
<td style="text-align:left;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_2
</td>
<td style="text-align:left;">
56
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_3
</td>
<td style="text-align:left;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_3
</td>
<td style="text-align:left;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_4
</td>
<td style="text-align:left;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_4
</td>
<td style="text-align:left;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Microglia_5
</td>
<td style="text-align:left;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Microglia_5
</td>
<td style="text-align:left;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
Granulocytes
</td>
<td style="text-align:left;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
APPPS1+
</td>
<td style="text-align:left;">
Granulocytes
</td>
<td style="text-align:left;">
16
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>variables=c(&quot;Celltype&quot;,&quot;Sex&quot;, &quot;Genotype_corr&quot;, &quot;Treatment&quot;,&quot;Phase&quot;, &quot;Brain_region&quot;,&quot;nCount_RNA&quot;,&quot;pseudoaligned_reads&quot;, &quot;percent.mito&quot;, &quot;percent.ribo&quot;, &quot;Mouse_ID&quot;)

res = compareGroups(Celltype~., data = samplemeta[,variables], max.ylev = 10)
#summary(res)
export_table &lt;- createTable(res)
export_table</code></pre>
<pre><code>
--------Summary descriptives table by &#39;Celltype&#39;---------

_____________________________________________________________________________________________________________________________________________________ 
                         T/NK       Microglia_0    Microglia_1    Microglia_2    Microglia_3    Microglia_4    Microglia_5    Granulocytes  p.overall 
                         N=22          N=186          N=164          N=107           N=99           N=26           N=23           N=22                
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
Sex:                                                                                                                                            .     
    f                 6 (27.3%)      27 (14.5%)     21 (12.8%)     11 (10.3%)     11 (11.1%)     1 (3.85%)      2 (8.70%)      8 (36.4%)              
    m                 16 (72.7%)    159 (85.5%)    143 (87.2%)     96 (89.7%)     88 (88.9%)     25 (96.2%)     21 (91.3%)     14 (63.6%)             
Genotype_corr:                                                                                                                               &lt;0.001   
    WT                3 (13.6%)      66 (35.5%)     93 (56.7%)     51 (47.7%)     44 (44.4%)     15 (57.7%)     11 (47.8%)     6 (27.3%)              
    APPPS1+           19 (86.4%)    120 (64.5%)     71 (43.3%)     56 (52.3%)     55 (55.6%)     11 (42.3%)     12 (52.2%)     16 (72.7%)             
Treatment:                                                                                                                                   &lt;0.001   
    Ctrl              3 (13.6%)     101 (54.3%)    113 (68.9%)     70 (65.4%)     68 (68.7%)     21 (80.8%)     12 (52.2%)     2 (9.09%)              
    Stroke            19 (86.4%)     85 (45.7%)     51 (31.1%)     37 (34.6%)     31 (31.3%)     5 (19.2%)      11 (47.8%)     20 (90.9%)             
Phase:                                                                                                                                        0.002   
    G1                1 (4.55%)      79 (42.5%)     70 (42.7%)     59 (55.1%)     42 (42.4%)     12 (46.2%)     14 (60.9%)     5 (22.7%)              
    G2M               11 (50.0%)     50 (26.9%)     38 (23.2%)     17 (15.9%)     21 (21.2%)     6 (23.1%)      4 (17.4%)      11 (50.0%)             
    S                 10 (45.5%)     57 (30.6%)     56 (34.1%)     31 (29.0%)     36 (36.4%)     8 (30.8%)      5 (21.7%)      6 (27.3%)              
Brain_region:                                                                                                                                &lt;0.001   
    Cortex            3 (13.6%)     101 (54.3%)    113 (68.9%)     70 (65.4%)     68 (68.7%)     21 (80.8%)     12 (52.2%)     2 (9.09%)              
    Lesion            19 (86.4%)     85 (45.7%)     51 (31.1%)     37 (34.6%)     31 (31.3%)     5 (19.2%)      11 (47.8%)     20 (90.9%)             
nCount_RNA          164049 (80916) 150451 (78955) 134224 (54802) 170420 (76702) 152640 (72678) 172200 (70772) 176087 (64299) 144828 (65899)   0.002   
pseudoaligned_reads 165078 (81748) 150942 (78850) 134355 (54810) 170750 (76639) 152790 (72668) 172433 (70717) 176390 (64225) 145316 (65972)   0.002   
percent.mito         2.23 (0.90)    1.75 (1.17)    1.62 (1.01)    2.07 (1.07)    1.72 (1.15)    1.97 (1.02)    2.11 (0.77)    0.92 (1.00)    &lt;0.001   
percent.ribo         6.53 (2.72)    2.68 (1.58)    2.75 (1.81)    2.44 (1.24)    3.28 (1.67)    2.11 (0.85)    2.91 (1.35)    2.01 (1.17)    &lt;0.001   
Mouse_ID:                                                                                                                                       .     
    23#15773          0 (0.00%)      7 (3.76%)      25 (15.2%)     12 (11.2%)     17 (17.2%)     4 (15.4%)      1 (4.35%)      0 (0.00%)              
    23#15774          0 (0.00%)      1 (0.54%)      30 (18.3%)     10 (9.35%)     6 (6.06%)      1 (3.85%)      1 (4.35%)      0 (0.00%)              
    23#15792          1 (4.55%)      11 (5.91%)     13 (7.93%)     9 (8.41%)      6 (6.06%)      6 (23.1%)      1 (4.35%)      0 (0.00%)              
    386               1 (4.55%)      14 (7.53%)     9 (5.49%)      2 (1.87%)      6 (6.06%)      1 (3.85%)      5 (21.7%)      3 (13.6%)              
    387               11 (50.0%)     10 (5.38%)     5 (3.05%)      6 (5.61%)      5 (5.05%)      0 (0.00%)      1 (4.35%)      6 (27.3%)              
    388               1 (4.55%)      41 (22.0%)     21 (12.8%)     18 (16.8%)     21 (21.2%)     5 (19.2%)      6 (26.1%)      0 (0.00%)              
    409               1 (4.55%)      33 (17.7%)     16 (9.76%)     18 (16.8%)     9 (9.09%)      3 (11.5%)      3 (13.0%)      3 (13.6%)              
    457               1 (4.55%)      41 (22.0%)     24 (14.6%)     21 (19.6%)     18 (18.2%)     5 (19.2%)      3 (13.0%)      2 (9.09%)              
    461               6 (27.3%)      28 (15.1%)     21 (12.8%)     11 (10.3%)     11 (11.1%)     1 (3.85%)      2 (8.70%)      8 (36.4%)              
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<pre class="r"><code>#get normalized counts 
# question to Desiree hat the Seurat object been initialized with normalized data?
counts &lt;- samples.integrated@assays$RNA@counts %&gt;% as.data.frame()

# drop no variance data and sort by samplemeta
counts &lt;- counts[apply(counts,1, sd) &gt; 0, rownames(samplemeta)]

# drop genes with low detection rate (more than 5 counts per cell)
counts_per_celltype=apply(counts, 1, function(x){tapply(x, samplemeta$Celltype, function(z){sum(z&gt;5,na.rm=T)})})

# keep RNAs with at least 5 cells with goood expression
idx=which(colSums(counts_per_celltype)&gt;5)

counts = counts[idx,]</code></pre>
</div>
<div id="hierarchical-clustering" class="section level2">
<h2>Hierarchical clustering</h2>
<p>to check where the variance in the data comes from</p>
<pre class="r"><code>log2_cpm = log2(counts+1)

varsset=apply(log2_cpm, 1, var)

cpm.sel.trans = t(log2_cpm[order(varsset,decreasing = T)[1:2000],])

distance = dist(cpm.sel.trans)

sampleDistMatrix &lt;- as.matrix(distance)

#colors for plotting heatmap
colors &lt;- colorRampPalette(brewer.pal(9, &quot;Spectral&quot;))(255)

cellcol = Dark8[1:nlevels(samplemeta$Celltype)]
names(cellcol) = levels(samplemeta$Celltype)

genotypecol = brewer.pal(4,&quot;Accent&quot;)[c(1:nlevels(samplemeta$Genotype_corr))]
names(genotypecol) = levels(samplemeta$Genotype_corr)

mousecol = brewer.pal(9,&quot;Set1&quot;)[1:nlevels(samplemeta$Mouse_ID)]
names(mousecol) = levels(samplemeta$Mouse_ID)

braincol = brewer.pal(3,&quot;Set2&quot;)[1:nlevels(samplemeta$Brain_region)]
names(braincol) = levels(samplemeta$Brain_region)

ann_colors = list(
  Genotype_corr = genotypecol, 
  Mouse_ID = mousecol,
  Brain_region = braincol,
  Celltype=cellcol
  )

labels = samplemeta[,c(&quot;Genotype_corr&quot;,&quot;Mouse_ID&quot;, &quot;Brain_region&quot;, &quot;Celltype&quot;)] %&gt;%  
  mutate_all(as.character) %&gt;% as.data.frame()

rownames(labels)=rownames(samplemeta)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = distance,
         clustering_distance_cols = distance,
         clustering_method = &quot;ward.D2&quot;,
         scale =&quot;row&quot;,
         show_rownames=F, show_colnames = F,
         legend=T,
         border_color = NA, 
         annotation_row = labels,
         annotation_col = labels,
         annotation_colors = ann_colors,
         col = colors, 
         main = &quot;D62 Distances normalized log2 counts&quot;)</code></pre>
<p><img src="figure/01_Biostat.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] pheatmap_1.0.12             RColorBrewer_1.1-2         
 [3] kableExtra_1.3.4            compareGroups_4.5.1        
 [5] data.table_1.14.2           SingleR_1.8.1              
 [7] SeuratObject_4.0.4          Seurat_4.1.0               
 [9] forcats_0.5.1               stringr_1.4.0              
[11] dplyr_1.0.7                 purrr_0.3.4                
[13] readr_2.1.2                 tidyr_1.1.4                
[15] tibble_3.1.6                ggplot2_3.3.5              
[17] tidyverse_1.3.1             DESeq2_1.34.0              
[19] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[21] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[23] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[25] IRanges_2.28.0              S4Vectors_0.32.3           
[27] BiocGenerics_0.40.0         limma_3.50.0               
[29] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] scattermore_0.7           bit64_4.0.5              
  [3] knitr_1.37                irlba_2.3.5              
  [5] DelayedArray_0.20.0       rpart_4.1.16             
  [7] KEGGREST_1.34.0           RCurl_1.98-1.5           
  [9] generics_0.1.2            ScaledMatrix_1.2.0       
 [11] callr_3.7.0               cowplot_1.1.1            
 [13] RSQLite_2.2.9             mice_3.14.0              
 [15] RANN_2.6.1                future_1.23.0            
 [17] chron_2.3-56              bit_4.0.4                
 [19] tzdb_0.2.0                spatstat.data_2.1-2      
 [21] webshot_0.5.2             xml2_1.3.3               
 [23] lubridate_1.8.0           httpuv_1.6.5             
 [25] assertthat_0.2.1          xfun_0.29                
 [27] hms_1.1.1                 jquerylib_0.1.4          
 [29] evaluate_0.14             promises_1.2.0.1         
 [31] fansi_1.0.2               dbplyr_2.1.1             
 [33] readxl_1.3.1              igraph_1.2.11            
 [35] DBI_1.1.2                 geneplotter_1.72.0       
 [37] Rsolnp_1.16               htmlwidgets_1.5.4        
 [39] spatstat.geom_2.3-1       ellipsis_0.3.2           
 [41] backports_1.4.1           annotate_1.72.0          
 [43] deldir_1.0-6              sparseMatrixStats_1.6.0  
 [45] vctrs_0.3.8               ROCR_1.0-11              
 [47] abind_1.4-5               cachem_1.0.6             
 [49] withr_2.4.3               HardyWeinberg_1.7.4      
 [51] sctransform_0.3.3         goftest_1.2-3            
 [53] svglite_2.0.0             cluster_2.1.2            
 [55] lazyeval_0.2.2            crayon_1.4.2             
 [57] genefilter_1.76.0         pkgconfig_2.0.3          
 [59] nlme_3.1-155              nnet_7.3-17              
 [61] rlang_1.0.0               globals_0.14.0           
 [63] lifecycle_1.0.1           miniUI_0.1.1.1           
 [65] modelr_0.1.8              rsvd_1.0.5               
 [67] cellranger_1.1.0          rprojroot_2.0.2          
 [69] polyclip_1.10-0           lmtest_0.9-39            
 [71] flextable_0.6.10          Matrix_1.4-0             
 [73] zoo_1.8-9                 reprex_2.0.1             
 [75] base64enc_0.1-3           whisker_0.4              
 [77] ggridges_0.5.3            processx_3.5.2           
 [79] png_0.1-7                 viridisLite_0.4.0        
 [81] bitops_1.0-7              getPass_0.2-2            
 [83] KernSmooth_2.23-20        Biostrings_2.62.0        
 [85] blob_1.2.2                DelayedMatrixStats_1.16.0
 [87] parallelly_1.30.0         beachmat_2.10.0          
 [89] scales_1.1.1              memoise_2.0.1            
 [91] magrittr_2.0.2            plyr_1.8.6               
 [93] ica_1.0-2                 zlibbioc_1.40.0          
 [95] compiler_4.1.2            fitdistrplus_1.1-6       
 [97] cli_3.1.1                 XVector_0.34.0           
 [99] listenv_0.8.0             patchwork_1.1.1          
[101] pbapply_1.5-0             ps_1.6.0                 
[103] MASS_7.3-55               mgcv_1.8-38              
[105] tidyselect_1.1.1          stringi_1.7.6            
[107] highr_0.9                 yaml_2.2.2               
[109] BiocSingular_1.10.0       locfit_1.5-9.4           
[111] ggrepel_0.9.1             grid_4.1.2               
[113] sass_0.4.0                tools_4.1.2              
[115] future.apply_1.8.1        parallel_4.1.2           
[117] rstudioapi_0.13           uuid_1.0-3               
[119] git2r_0.29.0              gridExtra_2.3            
[121] farver_2.1.0              Rtsne_0.15               
[123] digest_0.6.29             shiny_1.7.1              
[125] Rcpp_1.0.8                broom_0.7.12             
[127] later_1.3.0               writexl_1.4.0            
[129] RcppAnnoy_0.0.19          httr_1.4.2               
[131] gdtools_0.2.3             AnnotationDbi_1.56.2     
[133] colorspace_2.0-2          rvest_1.0.2              
[135] XML_3.99-0.8              fs_1.5.2                 
[137] tensor_1.5                reticulate_1.24          
[139] truncnorm_1.0-8           splines_4.1.2            
[141] uwot_0.1.11               spatstat.utils_2.3-0     
[143] plotly_4.10.0             systemfonts_1.0.3        
[145] xtable_1.8-4              jsonlite_1.7.3           
[147] R6_2.5.1                  pillar_1.7.0             
[149] htmltools_0.5.2           mime_0.12                
[151] glue_1.6.1                fastmap_1.1.0            
[153] BiocParallel_1.28.3       BiocNeighbors_1.12.0     
[155] codetools_0.2-18          utf8_1.2.2               
[157] lattice_0.20-45           bslib_0.3.1              
[159] spatstat.sparse_2.1-0     leiden_0.3.9             
[161] officer_0.4.1             zip_2.2.0                
[163] survival_3.2-13           rmarkdown_2.11           
[165] munsell_0.5.0             GenomeInfoDbData_1.2.7   
[167] haven_2.4.3               reshape2_1.4.4           
[169] gtable_0.3.0              spatstat.core_2.3-2      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
