<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scSeq_Hefendehl</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scSeq_Hefendehl/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltreedcc077d2096f036835bf35de2da79b6eea544995targetblankdcc077da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/dcc077d2096f036835bf35de2da79b6eea544995" target="_blank">dcc077d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscSeqHefendehltreedcc077d2096f036835bf35de2da79b6eea544995targetblankdcc077da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/tree/dcc077d2096f036835bf35de2da79b6eea544995" target="_blank">dcc077d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/20230203/
    Ignored:    data/Alignments/
    Ignored:    data/Cluster.markers_02-02-2023.xlsx
    Ignored:    data/Copy of DEG_control_vs_stroke_vs_distal.xlsx
    Ignored:    data/Copy of DEG_control_vs_stroke_vs_distal_backup.xlsx
    Ignored:    data/Copy of DEG_control_vs_stroke_vs_distal_original.xlsx
    Ignored:    data/DAM_genelists.RData
    Ignored:    data/DEG_control_vs_stroke.xlsx
    Ignored:    data/ReloadAllData_Hefendehl_Stroke_Dec'21.RData
    Ignored:    data/Sample_Tables/
    Ignored:    data/Stroke-SS2_SCT-harmony_Hefendehl.html
    Ignored:    data/counts.csv
    Ignored:    data/genecounts.csv
    Ignored:    data/microglia-SCT_02-02-23.rds
    Ignored:    data/microglia_protein.rds
    Ignored:    data/samples.integrated.RData
    Ignored:    data/seu-SCT-harmony_02-02-23.rds
    Ignored:    data/seu_filtered_nonorm_02-02-23.rds
    Ignored:    data/tx2genes.csv
    Ignored:    output/All_genes_Clusters.svg
    Ignored:    output/All_genes_Clusters_byCondition.svg
    Ignored:    output/All_genes_Clusters_byHomeostatgenes.svg
    Ignored:    output/All_genes_Clustertree.svg
    Ignored:    output/Celltypes_Opt_All_genes_all_cells_Clusters.svg
    Ignored:    output/Celltypes_Opt_All_genes_all_cells_Clusters_byCondition.svg
    Ignored:    output/Celltypes_Opt_All_genes_all_cells_Clusters_byPlate.svg
    Ignored:    output/Clusters_DEG_Markers2vs6.xlsx
    Ignored:    output/Clusters_DEG_Micorglia2vs4.xlsx
    Ignored:    output/Clusters_GOterms_DEG_Micorglia2vs4.xlsx
    Ignored:    output/Clusters_GOterms_DEG_Micorglia2vs6.xlsx
    Ignored:    output/Clusters_GOterms_Markers_all.xlsx
    Ignored:    output/Clusters_Markers_all.xlsx
    Ignored:    output/CrossFCplot.pdf
    Ignored:    output/CrossFCplot.svg
    Ignored:    output/Ellbowplot.svg
    Ignored:    output/G2M_Opt_All_genes_all_cells_Clusters.svg
    Ignored:    output/G2M_Opt_All_genes_all_cells_Clusters_byCondition.svg
    Ignored:    output/G2M_Opt_All_genes_all_cells_Clusters_byPlate.svg
    Ignored:    output/Gene_list_based_Clusters.svg
    Ignored:    output/Gene_list_based_Clusters_byCondition.svg
    Ignored:    output/Gene_list_based_Clusters_byHomeostatgenes.svg
    Ignored:    output/Gene_list_based_Clustertree.svg
    Ignored:    output/Microgila_WGCNA_Eigenscore.svg
    Ignored:    output/Microglia_DAM1_binary.svg
    Ignored:    output/Microglia_DAM1_bygenotype.svg
    Ignored:    output/Microglia_DAM1_calling.svg
    Ignored:    output/Microglia_DAM1_scoring.svg
    Ignored:    output/Microglia_DAM1_thresholding.svg
    Ignored:    output/Microglia_DAM2_bygenotype.svg
    Ignored:    output/Microglia_DAM2_calling.svg
    Ignored:    output/Microglia_DAM2proportionsof DAM1_per_condition_APPPS1.svg
    Ignored:    output/Microglia_DAM2proportionsof DAM1_per_condition_WT.svg
    Ignored:    output/Microglia_DAM2proportionsofDAM1_per_condition.svg
    Ignored:    output/Microglia_DAM2proportionsofDAM1_per_condition_APPPS1.svg
    Ignored:    output/Microglia_DAM2proportionsofDAM1_per_condition_WT.svg
    Ignored:    output/Microglia_DAMproportions_per_condition_APPPS1.svg
    Ignored:    output/Microglia_DAMproportions_per_condition_WT.svg
    Ignored:    output/Microglia_DAMproportions_per_condition_all.svg
    Ignored:    output/Microglia_DAMvsDAM2_contourplot.svg
    Ignored:    output/Microglia_DAMvsHomeo.svg
    Ignored:    output/Microglia_Dotplot_ClsuterCondition_Genelists.svg
    Ignored:    output/Microglia_HeatmapTop20DEG.svg
    Ignored:    output/Microglia_HomeoAndDAMproportions_per_condition_all.svg
    Ignored:    output/Microglia_Homeo_DAM.svg
    Ignored:    output/Microglia_Homeo_binary.svg
    Ignored:    output/Microglia_Homeo_bygenotype.svg
    Ignored:    output/Microglia_Homeo_calling.svg
    Ignored:    output/Microglia_Homeo_scoring.svg
    Ignored:    output/Microglia_Homeo_thresholding.svg
    Ignored:    output/Microglia_Homeoproportions_per_condition.svg
    Ignored:    output/Microglia_Homeoproportions_per_condition_APPPS1.svg
    Ignored:    output/Microglia_Homeoproportions_per_condition_WT.svg
    Ignored:    output/Microglia_HomeostatGenes_Cluster_Ridgeplot.svg
    Ignored:    output/Microglia_HomeostatGenes_Featureplot.svg
    Ignored:    output/Microglia_Opt_All_genes_Clusters.svg
    Ignored:    output/Microglia_Opt_All_genes_Clusters_byCondition.svg
    Ignored:    output/Microglia_Opt_All_genes_Clusters_byPlate.svg
    Ignored:    output/Microglia_Opt_All_genes_Condition.svg
    Ignored:    output/Microglia_Volcano_APPPS1CtrlvsAPPPS1Stroke.svg
    Ignored:    output/Microglia_Volcano_APPPS1strokevsWTStroke.svg
    Ignored:    output/Microglia_Volcano_WTCtrlvsAPPPS1ctrl.svg
    Ignored:    output/Microglia_Volcano_WTCtrlvsWTStroke.svg
    Ignored:    output/Microglia_Volcano_WT_StrokevsAPPPS1_Stroke.svg
    Ignored:    output/Microglia_Volcano_WT_strokevsAPPPS1Stroke.svg
    Ignored:    output/Microglia_Volcano_WTstrokevsAPPPS1Stroke.svg
    Ignored:    output/Microglia_WGCNA_Dendrocolplot.svg
    Ignored:    output/Microglia_logFCCtrlvsStroke_APPPS1vsWT.svg
    Ignored:    output/Microglia_logFCCtrlvsStroke_APPPS1vsWT_all.svg
    Ignored:    output/Microglia_logFCCtrlvsStroke_APPPS1vsWT_sig_only.svg
    Ignored:    output/Opt_All_genes_Clusters.svg
    Ignored:    output/Opt_All_genes_Clusters_byCondition.svg
    Ignored:    output/Opt_All_genes_Clusters_byPlate.svg
    Ignored:    output/Opt_All_genes_all_cells_Clusters.svg
    Ignored:    output/Opt_All_genes_all_cells_Clusters_byCondition.svg
    Ignored:    output/Opt_All_genes_all_cells_Clusters_byPlate.svg
    Ignored:    output/WGCNAGeneWise_ExpressionModuleblue.svg
    Ignored:    output/WGCNAGeneWise_ExpressionModulebrown.svg
    Ignored:    output/WGCNAGeneWise_ExpressionModulegreen.svg
    Ignored:    output/WGCNAGeneWise_ExpressionModuleturquoise.svg
    Ignored:    output/WGCNAGeneWise_ExpressionModuleyellow.svg
    Ignored:    output/WGCNA_labeled_Heatmap.svg
    Ignored:    output/cluster_desiree_orginal.pdf
    Ignored:    output/geneInfoSigned.csv
    Ignored:    output/integrated_analysis.RData
    Ignored:    output/integrated_analysis.RDataTmp
    Ignored:    output/net.rds
    Ignored:    output/net_from_tmp.rds
    Ignored:    output/powerTable_from_tmp.rds
    Ignored:    output/powertable.rds
    Ignored:    output/tmp.RData

Untracked files:
    Untracked:  .RData_backup
    Untracked:  ConsensusTOM-block.1.rda
    Untracked:  analysis/.blockConsModsCachec4391c51fd
    Untracked:  analysis/.blockConsModsCachec44e99f35
    Untracked:  analysis/.blockConsModsCachec46b55da8a
    Untracked:  analysis/.blockConsModsCachec46d30acb4
    Untracked:  analysis/ConsensusTOM-block.1.rda
    Untracked:  shinySecret.R
    Untracked:  wd_23-11-16.RData

Unstaged changes:
    Deleted:    analysis/02_plotsforPub.Rmd
    Modified:   code/custom_functions.R
    Modified:   scSeq_Hefendehl.Rproj
    Modified:   wflowhelper.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/dcc077d2096f036835bf35de2da79b6eea544995/analysis/index.Rmd" target="_blank">dcc077d</a>
</td>
<td>
achiocch
</td>
<td>
2024-02-09
</td>
<td>
workflowr::wflow_publish(c("./analysis/03<em>",
"./analysis/index</em>"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/2290b2325c4e78baccc13bb08837b714020f63db/analysis/index.Rmd" target="_blank">2290b23</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-11-03
</td>
<td>
backup 20231103
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/1f4af15ef5ceba8afc10bcdc4f42d65ded375b0e/docs/index.html" target="_blank">1f4af15</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-08-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/87fbacb6cfad2e04d135459ee7e2d02c2d1da414/docs/index.html" target="_blank">87fbacb</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/6d3e973239a1bcc44c8cec12decf1c235230ad0b/docs/index.html" target="_blank">6d3e973</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/982b9f6aab40e89a3a71620c77ffb6cd06495693/docs/index.html" target="_blank">982b9f6</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/438f7ada0b73e5f1c8f03701d236b7910912e760/analysis/index.Rmd" target="_blank">438f7ad</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-02-20
</td>
<td>
workflowr::wflow_publish(c("./docs/<em>", "./analysis/</em>",
"./code/*",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/0e20f22fd6f4bd1f2f5454bb9f515af001251bf2/docs/index.html" target="_blank">0e20f22</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/eaa373d44abba957ff4d3738a8d952454338988d/analysis/index.Rmd" target="_blank">eaa373d</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2023-02-16
</td>
<td>
workflowr::wflow_publish(c("./docs/<em>", "./analysis/</em>",
"./code/*",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/776448493ac55dc6d14eefa722acbb0652fe3dfc/docs/index.html" target="_blank">7764484</a>
</td>
<td>
achiocch
</td>
<td>
2022-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/14e1ee02e7bfea54ee17a0eb5498fe1a2db57c43/docs/index.html" target="_blank">14e1ee0</a>
</td>
<td>
achiocch
</td>
<td>
2022-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/3451e151c6a42a39cfed7acafdb1a79335186ff4/docs/index.html" target="_blank">3451e15</a>
</td>
<td>
achiocch
</td>
<td>
2022-06-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/cfbfcf64e976caa763142f22cfebdc2e47b49b43/analysis/index.Rmd" target="_blank">cfbfcf6</a>
</td>
<td>
achiocch
</td>
<td>
2022-06-22
</td>
<td>
wflow_publish(c("analysis/<em>", "docs/</em>", "code/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/cfbfcf64e976caa763142f22cfebdc2e47b49b43/docs/index.html" target="_blank">cfbfcf6</a>
</td>
<td>
achiocch
</td>
<td>
2022-06-22
</td>
<td>
wflow_publish(c("analysis/<em>", "docs/</em>", "code/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/faf20e9100d4b3d33fecbcf253b6ee1fde775ad7/docs/index.html" target="_blank">faf20e9</a>
</td>
<td>
achiocch
</td>
<td>
2022-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/51b35c896621dd205d47cc8c43580869f877acfa/docs/index.html" target="_blank">51b35c8</a>
</td>
<td>
achiocch
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/41d6cd0a5ec98105c1e404478f2c44913f0bc16e/docs/index.html" target="_blank">41d6cd0</a>
</td>
<td>
achiocch
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/30f02ebe4c9c504ef04566da21529ffd71f9694c/docs/index.html" target="_blank">30f02eb</a>
</td>
<td>
achiocch
</td>
<td>
2022-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/9b778b1461f069bd3cca3cb5f3ae21317fcb06cf/docs/index.html" target="_blank">9b778b1</a>
</td>
<td>
achiocch
</td>
<td>
2022-04-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/17114be918145efcd134fc88f90439aeeac0ed21/docs/index.html" target="_blank">17114be</a>
</td>
<td>
achiocch
</td>
<td>
2022-04-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/cf395f4c32e9a758af6bd5607b046cd8e779f0fe/docs/index.html" target="_blank">cf395f4</a>
</td>
<td>
achiocch
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/c998828253fd4d42e1e85fe5dd3102fc218710a6/docs/index.html" target="_blank">c998828</a>
</td>
<td>
achiocch
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/87439a3f88cf1423d6dd2da8e9d0f4cfac576aad/docs/index.html" target="_blank">87439a3</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/d6f2105cad0906fa053b2c10efe98ebb852a4099/docs/index.html" target="_blank">d6f2105</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/79dd0b17f11fa334e2736de456bdd1deb96bf3ee/docs/index.html" target="_blank">79dd0b1</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/ded601da88519316d8c1d645629bcef8acf346b4/docs/index.html" target="_blank">ded601d</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/50fe2116353a2ca9be2cfa503a99e7bd36282598/docs/index.html" target="_blank">50fe211</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/64cd37cb31ab6bb8935fdf508d12232202b9de90/analysis/index.Rmd" target="_blank">64cd37c</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-03
</td>
<td>
wflow_publish(c("analysis/<em>", "docs/</em>", "code/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/5fbdba4939eedb86fd65eaf1ebfd56f9398bbadd/docs/index.html" target="_blank">5fbdba4</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/90447988e7bd6d9b0640c3a3b22f3f5bb342877e/docs/index.html" target="_blank">9044798</a>
</td>
<td>
Andreas Geburtig-Chiocchetti
</td>
<td>
2022-02-02
</td>
<td>
workflowr automated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/57c9e7069d97adb6f27b2894ad255a42d62c9229/analysis/index.Rmd" target="_blank">57c9e70</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-02
</td>
<td>
adds initial html files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/57c9e7069d97adb6f27b2894ad255a42d62c9229/docs/index.html" target="_blank">57c9e70</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-02
</td>
<td>
adds initial html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/ddb63739581ad5617cc1e7ed6a5b994ff35b12e0/analysis/index.Rmd" target="_blank">ddb6373</a>
</td>
<td>
achiocch
</td>
<td>
2022-02-02
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/f010e791c623c3e27b3164c91bd3a890c2d194c8/docs/index.html" target="_blank">f010e79</a>
</td>
<td>
achiocch
</td>
<td>
2022-01-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scSeq_Hefendehl/4c4711a69dbc290774b236d403203b2a0d479705/docs/index.html" target="_blank">4c4711a</a>
</td>
<td>
achiocch
</td>
<td>
2022-01-31
</td>
<td>
wflow_publish(c("analysis/<em>", "docs/</em>", "code/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scSeq_Hefendehl/blob/730bee8286adbc4b9d9339d7442cd69e4cb427ed/analysis/index.Rmd" target="_blank">730bee8</a>
</td>
<td>
achiocch
</td>
<td>
2022-01-31
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here you find all analyses, results and visualizations</p>
<ul>
<li><p>Frist analysis provided by Desiree <a
href="00_Original_Analysis.html">view</a></p></li>
<li><p>Follow up analysis with new WT data <a
href="00.1_Original_Analysis_Integrated.html">view</a> <br><em>analysis
adapted from originalscript of Desiree</em></p></li>
<li><p>Statistical analysis <a href="01_Biostat.html">view</a> <em>based
on new WT data</em></p></li>
<li><p>Analyses for Publication <a
href="03_New_AnalysesforPub.html">view</a> <em>based on Wishlist Sept
2023, includes new preprocessing, fitlering and clustering of
cells</em></p></li>
<li><p>Plots for Publication <a
href="04_plotsforPub.html">view</a></p></li>
</ul>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
