---
title: "Biostatistical analysis part 2"
author: "AGC"
date: "2 2 2022"
output: html_document
---

```{r setup, include=F}
knitr::opts_chunk$set(echo = TRUE, fig.height=8, fig.width=15)

library(limma)
library(DESeq2)
library(tidyverse)
library(Seurat)
library(SingleR)
library(data.table)
library(compareGroups)
library(brms)
library(xlsx)
library(viridis)
library(DT)
library(slingshot)
library(SingleCellExperiment)

load("./output/integrated_analysis.RData")
source("./code/custom_functions.R")

trimmed_scaled <- function(vec){
  m <- max(vec, na.rm=T)
  vec[vec>quantile(vec,.95)] = m
  return(scale(vec))
}


```


```{r Load Cellproperties}

#get sample data
samples.integrated@meta.data %>% as.data.frame() -> samplemeta


# convert to correct data type 

# define genotype as is
samplemeta$Genotype_corr = factor(samplemeta$Genotype=="WT", levels=c(F,T), labels = c("APPPS1+", "WT"))
samplemeta$Genotype_corr = relevel(samplemeta$Genotype_corr, ref="WT")

samplemeta$methoxy = factor(samplemeta$Methoxy=="pos", levels=c(T,F), labels = c("MX04+", "MX04-"))

samplemeta$Treatment = as.factor(samplemeta$Treatment)
samplemeta$Treatment = relevel(samplemeta$Treatment, ref="Ctrl")
samplemeta$Mouse_ID = as.factor(samplemeta$Mouse_ID)
samplemeta$Sex = as.factor(samplemeta$Gender)
samplemeta$Brain_region = as.factor(samplemeta$Brain_region)
samplemeta$Celltype = as.factor(samplemeta$Celltype)

samples.integrated@meta.data = samplemeta

nCells=nrow(samplemeta)
nMice=nlevels(samplemeta$Mouse_ID)
nCelltypes=nlevels(samplemeta$Celltype)


```

## Sample descriptive: 

Data  contains a total of `r nCells` Cells from `r nMice`. 

**Cells per Mouse**

Cells per Strain

```{r table strains}
table(Mouse_ID=samplemeta$Mouse_ID) %>% as.data.frame() %>% display_tab()

```

**Cells per Genotype**

```{r Cells per Strain}
table(Genotype=samplemeta$Genotype_corr, Treatment=samplemeta$Treatment) %>% as.data.frame() %>% display_tab()

table(Genotype=samplemeta$Genotype_corr, Celltype=samplemeta$Celltype) %>% 
  as.data.frame() %>% display_tab()


table(Genotype=samplemeta$Genotype_corr, 
      Celltype=samplemeta$Celltype, Treatment=samplemeta$Treatment
) %>% 
  as.data.frame() %>% display_tab()

samplemeta$condition=paste0(samplemeta$Genotype_corr,"_", samplemeta$Treatment)
samples.integrated@meta.data = samplemeta

sce=as.SingleCellExperiment(samples.integrated)

sce <- slingshot(sce, clusterLabels = 'Celltype', reducedDim = 'UMAP')
samplemeta$Pseudotime <- sce$slingPseudotime_1
samples.integrated@meta.data = samplemeta

variables=c("Celltype","Sex", "Age","Genotype_corr", "Treatment", "condition","Phase", "Pseudotime","Brain_region","nCount_RNA","pseudoaligned_reads", "percent.mt", "percent.rp", "Mouse_ID")

```

Descriptive stats across Cell type
```{r Compare Celltype}
res = compareGroups(Celltype~., data = samplemeta[,variables], max.ylev = 10)
#summary(res)
export_table <- createTable(res)
options(width = 10000)
export2md(export_table)
export2xls(export_table,paste0(home,"/docs/Descriptives.xlsx"))

```

download data as excel file [here](Descriptives.xlsx)

## Preprocessing 

include only genes that are different across samples (5 974 excluded)
include only genes with more than 10 reads in at least 10 cells in at least one Celltype (21 733 transcripts excluded)
exclude all cells with less than 10000 reads (none excluded)

649 cells and 11 623 transcripts analyzed

```{r read gene counts}

#get normalized counts 
# question to Desiree hat the Seurat object been initialized with normalized data?
counts <- samples.integrated@assays$RNA@counts %>% as.data.frame()

# drop no variance data and sort by samplemeta
counts <- counts[apply(counts,1, sd) > 0, rownames(samplemeta)]

# drop genes with low detection rate (more than 5 counts per cell)
counts_per_celltype=apply(counts, 1, function(x){tapply(x, samplemeta$Celltype, function(z){sum(z>10,na.rm=T)})}) %>% as.data.frame()

# keep RNAs with at least 10 cells with good expression
idxr=which(colSums(counts_per_celltype)>10)

idxc=which(colSums(counts)>10000)

counts = counts[idxr,idxc]
samplemeta=samplemeta[colnames(counts), ]

```

## Overview tables
```{r summary tables}

samplemeta$corrGenotype_Treatment<-
  paste0(samplemeta$Genotype_corr," ", samplemeta$Treatment)

sumstat = counts %>% t() %>%  
  aggregate(by=samplemeta["corrGenotype_Treatment"], mean) %>% 
  t() %>% as.data.frame()

names(sumstat) <- paste0("mean",sumstat[1,])
sumstat<-sumstat[-1,]

sumstatCelltype = counts %>% t() %>%  
  aggregate(by=samplemeta["Celltype"], mean) %>% 
  t() %>% as.data.frame()

names(sumstatCelltype) <- paste0("mean",sumstatCelltype[1,])
sumstatCelltype<-sumstatCelltype[-1,]


Geno.Treatment_kruskal=apply(counts,1, function(x){
  res = kruskal.test(unlist(x)~samplemeta$corrGenotype_Treatment) %>% unlist()
  return(as.numeric(res["p.value"]))
})

Geno.Treatment_kruskal_fdr = p.adjust(Geno.Treatment_kruskal, method = "fdr")

Celltype_kruskal=apply(counts,1, function(x){
   res = kruskal.test(unlist(x)~samplemeta$Celltype) %>% unlist()
  return(as.numeric(res["p.value"]))
})

Celltype_kruskal_fdr = p.adjust(Celltype_kruskal, method = "fdr")

summarytab=cbind(sumstat, Geno.Treatment_kruskal, Geno.Treatment_kruskal_fdr , sumstatCelltype, Celltype_kruskal, Celltype_kruskal_fdr)

summarytab %>% display_tab()

```



## Hierarchical clustering 

to check where the variance in the data comes from 

```{r basic QC, fig.height=10, fig.width=10}
log2_cpm = log2(counts+1)

varsset=apply(log2_cpm, 1, var)

cpm.sel.trans = t(log2_cpm[order(varsset,decreasing = T)[1:1000],])

distance = dist(cpm.sel.trans)

sampleDistMatrix <- as.matrix(distance)

#colors for plotting heatmap
colors <- rev(colorRampPalette(brewer.pal(9, "Spectral"))(255))
colors=jetcolors(255)
colors=viridis(255)

cellcol = Dark8[1:nlevels(samplemeta$Celltype)]
names(cellcol) = levels(samplemeta$Celltype)

genotypecol = brewer.pal(4,"Accent")[c(1:nlevels(samplemeta$Genotype_corr))]
names(genotypecol) = levels(samplemeta$Genotype_corr)

strokecol = brewer.pal(5,"Set2")[1:nlevels(samplemeta$Treatment)+2]
names(strokecol) = levels(samplemeta$Treatment)

mousecol = brewer.pal(9,"Set1")[1:nlevels(samplemeta$Mouse_ID)]
names(mousecol) = levels(samplemeta$Mouse_ID)

braincol = brewer.pal(3,"Set2")[1:nlevels(samplemeta$Brain_region)]
names(braincol) = levels(samplemeta$Brain_region)

samplemeta$Age = as.factor(samplemeta$Age)
Agecol = colorRampPalette(c("dodgerblue",
                          "dodgerblue4"))(nlevels(samplemeta$Age))[1:nlevels(samplemeta$Age)]
names(Agecol) = levels(samplemeta$Age)

ann_colors = list(
  Genotype_corr = genotypecol, 
  Mouse_ID = mousecol,
  Brain_region = braincol,
  Celltype=cellcol,
  Treatment=strokecol, 
  Age=Agecol
)

labels = samplemeta[,c("Genotype_corr","Mouse_ID", "Age", "Brain_region", "Celltype", "Treatment")] %>%  
  mutate_all(as.character) %>% as.data.frame()
rownames(labels)=rownames(samplemeta)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = distance,
         clustering_distance_cols = distance,
         clustering_method = "ward.D2",
         scale ="none",
         legend = F,
         show_rownames=F, show_colnames = F,
         border_color = NA, 
         annotation_row = labels,
         annotation_col = labels,
         annotation_colors = ann_colors,
         col = colors, 
         main = "D62 top1000 Distances normalized log2 counts")

```

```{r heatmap_legend, fig.height=3, fig.width=10, echo=F}
heatmaplegend=function(){
  plot(0,0,type='n',
     axes=FALSE,ann=FALSE, 
     xlim=c(0,1), ylim=c(0,1))
title(main="legend")

grid=1:length(ann_colors)/length(ann_colors)

for(i in 1:length(ann_colors))
  legend(x = grid[i],y=1, xpd=T,xjust = 1,
         title = names(ann_colors[i]),bty = "n",
         legend=names(ann_colors[[i]]),pch=15, 
         col = ann_colors[[i]])
}

heatmaplegend()

```

## Genotype X Treatment Statistical modelling

the logFC should not be interpreted on its own without the specific post hoc tests (see app below to check the individual genes) In any case the value here would correspond to the APPPS1+ with Stroke against all others.

Heatmaps show standardized deviations from mean across all cells, trimmed to 2 standard deviations ( i.e. values above 2 SD are set to 2 SD). this allows to see more subtle changes better. 



```{r define LMER function}

getres=function(Celltype="Specify", 
                Hypothesis="~1+Sex+Genotype_corr*Treatment", 
                Target="Genotype_corrAPPPS1+:TreatmentStroke",
                Randomeffect=NULL){
  
  res= comparison_rand(designform =Hypothesis, 
                       randomeffect = Randomeffect, 
                       Samples = samplemeta$Celltype==Celltype, 
                       log_cpm = log2_cpm, 
                       samplesdata = samplemeta,
                       target=Target)
  
  
  
  labels = samplemeta[samplemeta$Celltype==Celltype,c("Treatment","Age", "Genotype_corr","Mouse_ID", "Brain_region", "Celltype")] %>%  
    mutate_all(as.character) %>% as.data.frame()

  
  labels=labels %>% arrange(Treatment,Genotype_corr)
  plotdata=log2_cpm[res$adj.P.Val<0.05,rownames(labels)]
  plotdata = apply(plotdata, 2, trimmed_scaled)
  
  
  
  pheatmap(plotdata,
           #clustering_method = "ward.D2",
           cluster_cols = F,
           cluster_rows  = T,
           scale ="row",
           show_rownames=F, show_colnames = F,
           legend=T,
           border_color = NA, 
           #annotation_row = labels,
           annotation_col = labels,
           annotation_colors = ann_colors,
           col = colors, 
           breaks = seq(-2,2, length.out=254),
           main = "D62 Distances normalized log2 counts")
  return(res)
  
}

generate_output=function(CT="Celltype", ...){
  respath=paste0(home, "/docs/LMER_",CT,".xlsx")
  
  analysis = getres(Celltype = CT)
  analysis_sig = analysis[analysis$adj.P.Val<0.05,]
  analysis_sig %>% display_tab()
  
  write.xlsx2(analysis_sig, file =respath , sheetName = "significant genes")
  
  ResGO = getGOresults(rownames(analysis_sig),
                       rownames(analysis),
                       "mmusculus")
  
  if(length(ResGO)>0){
    p=gostplot(ResGO)
  } else{
    ResGO=data.frame(result="no significant enrichment identified")
    p="no significant enrichment identified"
  }
  
  write.xlsx2(ResGO$result, file = respath, sheetName = "GO_enrichment", append=T)
  return(list(results=analysis,results_sig=analysis_sig, plot=p))
}

```


### Microglia_1
```{r Microglia_1, message=F, warning=FALSE}
res_output=generate_output(CT = "Microglia_1")
res_output[["results_sig"]] %>% display_tab()
res_output[["plot"]]
```
Data sources and their abbreviations are: [Gene Ontology](http://geneontology.org/) (GO or by branch GO:MF, GO:BP, GO:CC)[KEGG](https://www.genome.jp/kegg/) (KEGG) [TRANSFAC](https://genexplain.com/transfac/) (TF) [miRTarBase](https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2019/php/index.php) (MIRNA) [CORUM](https://mips.helmholtz-muenchen.de/corum/) (CORUM) [Human phenotype ontology](http://hpo.jax.org/app/) (HP) [Human Protein Atlas](https://www.proteinatlas.org/) (HPA)

get results table [here](LMER_Microglia_1.xlsx)


### Microglia_2
```{r Microglia_2, message=F, warning=FALSE}
res_output=generate_output("Microglia_2")
res_output[["results_sig"]] %>% display_tab()
res_output[["plot"]]
```
Data sources and their abbreviations are: [Gene Ontology](http://geneontology.org/) (GO or by branch GO:MF, GO:BP, GO:CC)[KEGG](https://www.genome.jp/kegg/) (KEGG) [TRANSFAC](https://genexplain.com/transfac/) (TF) [miRTarBase](https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2019/php/index.php) (MIRNA) [CORUM](https://mips.helmholtz-muenchen.de/corum/) (CORUM) [Human phenotype ontology](http://hpo.jax.org/app/) (HP) [Human Protein Atlas](https://www.proteinatlas.org/) (HPA)

get results table [here](LMER_Microglia_2.xlsx)

### Microglia_3
```{r Microglia_3, message=F, warning=FALSE}
res_output=generate_output("Microglia_3")
res_output[["results_sig"]] %>% display_tab()
res_output[["plot"]]
```
Data sources and their abbreviations are: [Gene Ontology](http://geneontology.org/) (GO or by branch GO:MF, GO:BP, GO:CC)[KEGG](https://www.genome.jp/kegg/) (KEGG) [TRANSFAC](https://genexplain.com/transfac/) (TF) [miRTarBase](https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2019/php/index.php) (MIRNA) [CORUM](https://mips.helmholtz-muenchen.de/corum/) (CORUM) [Human phenotype ontology](http://hpo.jax.org/app/) (HP) [Human Protein Atlas](https://www.proteinatlas.org/) (HPA)

get results table [here](LMER_Microglia_3.xlsx)

### Microglia_4
calculation not converging as Celltype not identified in all conditions 
however also rare in other conditions, no significant difference 

```{r Microglia_4, message=F, warning=FALSE}
idx=samplemeta$Celltype=="Microglia_4"
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])

table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %>% fisher.test()

#res_output=generate_output("Microglia_4")
#res_output[["results_sig"]] %>% display_tab()
#res_output[["plot"]]

```

get results table [here](LMER_Microglia_4.xlsx)

### Microglia_5
```{r Microglia_5, message=F, warning=FALSE}
res_output=generate_output("Microglia_5")
res_output[["results_sig"]] %>% display_tab()
res_output[["plot"]]

```
Data sources and their abbreviations are: [Gene Ontology](http://geneontology.org/) (GO or by branch GO:MF, GO:BP, GO:CC)[KEGG](https://www.genome.jp/kegg/) (KEGG) [TRANSFAC](https://genexplain.com/transfac/) (TF) [miRTarBase](https://mirtarbase.cuhk.edu.cn/~miRTarBase/miRTarBase_2019/php/index.php) (MIRNA) [CORUM](https://mips.helmholtz-muenchen.de/corum/) (CORUM) [Human phenotype ontology](http://hpo.jax.org/app/) (HP) [Human Protein Atlas](https://www.proteinatlas.org/) (HPA)
get results table [here](LMER_Microglia_5.xlsx)


### Granulozytes
calculation not converging as Celltype not identified in all conditions 
however also rare in other conditions, no significant difference 
```{r Granulozytes, message=F, warning=FALSE}

idx=samplemeta$Celltype=="Granulocytes"
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %>% fisher.test()

# res_output=generate_output("Granulocytes ")
# res_output[["results_sig"]] %>% display_tab()
# res_output[["plot"]]

```

get results table [here](LMER_Granulozytes.xlsx)

### Mono/Mac

```{r T_NK, message=F, warning=FALSE}

idx=samplemeta$Celltype=="Mono/Mac"
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx])
table(samplemeta$Treatment[idx], samplemeta$Genotype_corr[idx]) %>% fisher.test()
# res_output=generate_output("T/NK")
# res_output[["results_sig"]] %>% display_tab()
# res_output[["plot"]]

```

get results table [here](LMER_T_NK.xlsx)

## Pseudotime analysis 

```{r calculate pseudotime}


colors <- viridis(100)
plotcol <- colors[cut(sce$slingPseudotime_1, breaks=100)]

plot(reducedDims(sce)$UMAP, col = "grey", pch=16, asp = 1)
points(reducedDims(sce)$UMAP, col = plotcol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, col='black', type="l")



plot(samplemeta$Pseudotime[samplemeta$condition == "APPPS1+_Ctrl"] %>% density(na.rm=T), col=Dark8[4], lwd=2, xlab="Pseudotime", main="distribution of cells over pseudotime")
lines(samplemeta$Pseudotime[samplemeta$condition == "APPPS1+_Stroke"] %>% density(na.rm=T), col=Dark8[2], lwd=2)
lines(samplemeta$Pseudotime[samplemeta$condition == "WT_Stroke"] %>% density(na.rm=T), col=Dark8[3], lwd=2)
lines(samplemeta$Pseudotime[samplemeta$condition == "WT_Ctrl"] %>% density(na.rm=T), col=Dark8[1], lwd=2)
legend("topright", legend=c("WT_Ctrl","APPPS1+_Stroke","WT_Stroke","APPPS1+_Ctrl"), fill = Dark8[1:4])

```



## Gene expression plot

```{r setapp, include=F}

secret="/files/scSeq_Hefendehl/shinySecret.R"
source(secret)

local({
  r <- getOption("repos")
  r["CRAN"] <- "https://cran.rstudio.com/"
  r["BioCsoft"] <- "https://bioconductor.org/packages/release/bioc"
  r["BioCann"] <- "https://bioconductor.org/packages/release/data/annotation"
  r["BioCexp"] <- "https://bioconductor.org/packages/release/data/experiment"
  options(repos = r)
})

counts=as.data.frame(samples.integrated@assays$RNA@data)
counts = counts[,rownames(samplemeta)]
samplemeta= cbind(samplemeta, samples.integrated@reductions$tsne@cell.embeddings[row.names(samplemeta), ])
samplemeta= cbind(samplemeta, samples.integrated@reductions$umap@cell.embeddings[row.names(samplemeta), ])


save(list=c("samplemeta", "counts"), file=paste0(home, "/geneviewer/Dataset.RData"))



rsconnect::setAccountInfo(name='molgenlab',
                          token='86875F8B6550C3A26488035E69B1F18D',
                          secret=shinySECRET)


rsconnect::deployApp(paste0(home, "/geneviewer"), forceUpdate = T)

```


<iframe src="https://molgenlab.shinyapps.io/geneviewer/" height=900px width=900px></iframe>

<button onclick="window.location.href='https://molgenlab.shinyapps.io/geneviewer/'">Open in Browser</button>

<p></p>

<p></p>
